{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Glick Ch3</h6>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "<h6>Produce DNA w/o cloning, Gibson Assembly</h6>\n",
    "https://en.wikipedia.org/wiki/Gibson_assembly\n",
    "    \n",
    "How is gibson assembly different from PCR? \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "<h6>Transormation and Selection</h6>\n",
    "The process of uptake of the designed plasmid into Ecoli is called transformation. A cell which is capable of\n",
    "taking in this DNA is called competent. When cell density is high or starvation impending bacterial cells will produce\n",
    "proteins which aid in uptake of plasmids. Transformation defined as 1) binding of double stranded DNA to cell wall, \n",
    "2) entroof DNA into inner compartment, periplasm, 3) transmission of 1 strand into the cytoplasm while the other\n",
    "is degraded, 4) if hte DNA is linear integration into the host chromosome. Competence can be induced by CaCl2 and \n",
    "heat shock. If the introduced DNA is a plasmid, it is maintained in the cytoplasm until the second strand is \n",
    "synthesized. Transformation efficience defined as number of transformed cells as funciton of amount of DNA added.\n",
    "Transformation efficiency is approx 10^7 tp 10^8 transformed cells per microgram of intact plasmid DNA. \n",
    "Transformation\n",
    "inefficeint, about 1/1k cells are transformed. \n",
    "<li>Bacteria which are RecA- have higher prob of success</li>\n",
    "<li>Bacteria that dont produce endonulcease encoded by endA1 gene have higher tranformation frequencies</li>\n",
    "For selection, 2 antibiotic resistant genes are insterted into the plasmid so after putting in plate w/Antibiotics\n",
    "the ones which took in the plasmid should survive. \n",
    "<img src=\"t1.png\">\n",
    "\n",
    "<img src=\"t2.png\">\n",
    "<img src=\"t3.png\">\n",
    "<img src=\"t4.png\">\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Other bacteria like Bacillus subtilis and Agrobacterium tumefaciens act as a final host cells. Cloning \n",
    "vectors that funciton in EColi can have a second ORI to enable the plasmid to replicate in a separate\n",
    "host cell! Shuttle vectors are vectors which can operate in 2 different hosts. Prokaryotes dont have intons, Eukaryotes\n",
    "do. There are some shuttle vectors which can operate in both. To create a shuttle vector, start with an Ecoli plasmid then \n",
    "combine the Ecoli plasmid part with the insert and antibiotic resistence gene with part of the host cell specific\n",
    "plasmid with a origin of replication. This ORI is different than the Ecoli plasmid ORI. First step to make Shuttle\n",
    "vector is to engineer 2 recognition sites for the restriction endonuclease SfiI. The seque3nce is GGCCNNNNNGGCC/CCGGNNNNNCCGG.\n",
    "2 different sites SfiIx and SfiIy have 2 different sequences so after digestion with SFiI teh extesions of SfiIx \n",
    "are not complimentary to SfiIy. Then insert teh 2 SfiI sequences so after digestion they flank teh region containing\n",
    "the antibiotic resistence gene adn cloing site on the EColi plasmid and the ORI on the host. The SfiIx-SfiIy procedure\n",
    "can be adopted for any host cell that has a plasimid is called vector backbone exchange. \n",
    "<img src=\"g1.png\">\n",
    "Making a genomic library. Prokaryotes no intons. Eukaryotes have intons. Different genomic library prep procedures \n",
    "for both cell types. \n",
    "<img src=\"g2.png\">\n",
    "number of DNA fragments should be 3x size of DNA genome when cutting DNA for genomic library. Create a partial digestion\n",
    "where you get a distribution of all possible cuts and fragment sizes. Assuming this is an even distribution. \n",
    "$$3\\frac{gene length}{fragment size}=3\\frac{4*10^6}{10^3)=12000$$ for genome 4*10^6 and fragnent size 1k. \n",
    "DNA library clone equation\n",
    "$$N = \\frac{ln(1-P)}{ln(1-f)}$$ Proof in the discussion slides\n",
    "<img src=\"g3.png\">\n",
    "This is a partial digestion vs. a full digestion. The red arrows show the places where the DNA is cut\n",
    "by the Restriction endonuclease, is you extend the time you get a wider distribution of smaller fragments. \n",
    "Can play with temp also in addition to heat. Reducing heat decreases effectiveness of RE. \n",
    "<img src=\"g4.png\">\n",
    "Table for genomic size. \n",
    "<img src=\"g5.png\">\n",
    "Once you have the DNA inserted in the clones; and they are replicated. How to filter out the ones you \n",
    "need? Denature DNA w/heat and RAPIDLY cool the single strands stay single. If the temp is lowered quickly\n",
    "the single strands will start to base pair. Heating and SLOWLY cooling called annealing. When DNA from different\n",
    "sources are annealed different source DNA will mix causing hybrids. We use this method to attach probe DNA to the sample\n",
    "    target DNA. We label the probe DNA with either flourscent or radio tag markers. Hybrid assay bc/combining from different\n",
    "                                                    soures, probe DNA+clone DNA\n",
    "                                                    \n",
    "<img src=\"g6.png\">\n",
    "Another method is the random primer method where you make all possible combinations of a 6 N sequence. Some of those will\n",
    "    hybridize to teh target dna. UNCLEAR HOW THIS WORKS>..\n",
    "<img src=\"g6a.png\">              \n",
    "                                                    \n",
    "<img src=\"g7.png\">\n",
    "<img src=\"g8.png\">\n",
    "<img src=\"g9.png\">\n",
    "<img src=\"g10.png\">\n",
    "<img src=\"g11.png\">\n",
    "<img src=\"g12.png\">\n",
    "<img src=\"g13.png\">\n",
    "<img src=\"g14.png\">\n",
    "<img src=\"g15.png\">\n",
    "<img src=\"g16.png\">\n",
    "<img src=\"g17.png\">\n",
    "<img src=\"g18.png\">\n",
    "<img src=\"g19.png\">\n",
    "<img src=\"g20.png\">\n",
    "Electroporation: to increase uptake of DNA/plasmid; subject bacteria to high voltage electric field. DNA is negatively\n",
    "charged, ecoli cells at 50 uliters are placed in chamger and pulse 20mF, 2.5kV and 200ohms for 4.6ms. This yields 10^9\n",
    "                                                transformants/ug of DNA for small plasmids ~3kb and 10^6 for large plasmids(~136kb)\n",
    "<img src=\"g21.png\">\n",
    "<img src=\"g22.png\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"p1.png\">\n",
    "<img src=\"p2.png\">\n",
    "<img src=\"p3.png\">\n",
    "<img src=\"p4.png\">\n",
    "<img src=\"p5.png\">\n",
    "<img src=\"p6.png\">\n",
    "<img src=\"p7.png\">\n",
    "<img src=\"p8.png\">\n",
    "<img src=\"p9.png\">\n",
    "<img src=\"p10.png\">\n",
    "<img src=\"p11.png\">\n",
    "<img src=\"p12.png\">\n",
    "<img src=\"p13.png\">\n",
    "<img src=\"p14.png\">\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    ". What are type II restriction endonucleases? \n",
    "Why are they important for recombinant DNA technology? \n",
    "2.  When circular double-stranded DNA from the plasmid pCEL1 is digested by various restriction endonucleases \n",
    "and combinations thereof, the following bands (with lengths given in kilobase pairs) are observed: \n",
    "    EcoRI, 6.0; BamHI, 6.0; HindIII, 6.0; HaeII, 3.0, 2.0, and 1.0; EcoRI and HaeII, 2.0 and 1.0; EcoRI and \n",
    "    HindIII, 3.5 and 2.5; EcoRI and BamHI, 4.5 and 1.5; BamHI and HindIII, 5.0 and 1.0; BamHI and \n",
    "    HaeII, 3.0, 1.5, 1.0, and 0.5; and HindIII and HaeII, 3.0, 1.5, 1.0, and 0.5. \n",
    "    Construct a restriction endonuclease enzyme site map with this information. \n",
    "3.  Describe how plasmid pBR322 is used as a cloning vector. What are its special features? \n",
    "4.  Describe the principal features of the pUC cloning system. \n",
    "5. A genomic library of a prokaryotic organism is often constructed by cloning the products of a \n",
    "Sau3AI partial digest of the genomic DNA into a BamHI site of the vector. \n",
    "a.  Why are two different enzymes used in this experiment? \n",
    "b. What is a partial digestion, and how is it performed? \n",
    "c.  Why is partial digestion often used for constructing genomic libraries? \n",
    "6.  For gene-cloning experiments, why is the cleaved plasmid DNA often treated with alkaline phosphatase \n",
    "prior to the ligation step? 7.  Suggest a few different ways that recombinant plasmids can be introduced into \n",
    "a gram-negative bacterium, such as E. coli . \n",
    "8.  Outline some of the strategies that are used to detect \n",
    "a cloned target gene within a library in E. coli . \n",
    "What conditions must be satisfied for each type of assay? \n",
    "9. What is a cDNA library? \n",
    "10.  Why would you use a plasmid, bacteriophage Î» , cosmid, or \n",
    "BAC as a cloning vector? \n",
    "11. What is a common mode of action of a type IIS restriction endonuclease? \n",
    "12. What is replica plating? Outline the basic methodology for replica plating. \n",
    "13. What is a fosmid? \n",
    "14. What is a tripartite mating? \n",
    "15. What is the role of RNase H during cDNA synthesis?"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
