{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "<h6>Review Transcription Regulation</h6>\n",
    "https://www.khanacademy.org/test-prep/mcat/biomolecules/gene-control/v/regulation-of-transcription\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "<h6>Glick Chapter2--Gene Expression</h6>\n",
    "Gene expression is the creation of proteins for a specific function starting from genes in the DNA. \n",
    "The chemical structure of DNA is:\n",
    "<li> 1) phosphate group</li>\n",
    "<li> 2) 5 carbon sugar or pentose or deoxyribose sugar. </li> \n",
    "<li>  3) base. </li>\n",
    "The base is {ATCG}, shown in the \n",
    "lower part of the figure connects via the N to the 1' carbon. DNA has a deoxyribose sugar, RNA a ribose sugar. \n",
    "Ribose sugar has 1 O attached to the 2' carbon. Deoxyribose does not have this O. \n",
    "<img src=\"glick1.png\">\n",
    "During DNA synthesis where there is are 2 single strand separated from a double strand and we are adding\n",
    "nucleotides to form a complimentary strand, the incoming nucleotide has 3 phosphate groups. The 3 phosphate\n",
    "groups are important because the covalent bonds are where the stored potential energy is. These phosphate bonds are\n",
    "in contrast to the peptide amine bonds during protein synthesis. There is a marked contrast between the DNA P bonds\n",
    "and the amide protein bonds. DNA is high potential energy easy to break w/enzyumes, proteins are harder to break, are more\n",
    "much more energy stable.\n",
    "<img src=\"glick3.png\">\n",
    "When the incoming nucleotide binds into the single strand to form a double strand 2 of the phosphate groups\n",
    "are lost and the remaining phoshpate group binds via the OH bond to the other base. This is in additon to a \n",
    "double H bond for A:T or triple H bond for C:G. The binding of the phosphate group to the pentose is called \n",
    "a phosphodiester bond. These phosphodiester bonds are in used when both RNA and DNA are synthesizing \n",
    "nucleotide sequences. \n",
    "<img src=\"glick2.png\">\n",
    "<img src=\"glick4.png\">\n",
    "shows the 2 P groups being jettisoned when the new phosphodiester bond forms. The bond releases energy and \n",
    "uses an enzyme(DNA Polymerase) to catalyze the reaction. There is negative charge keeping separation between the 2 dna strands. \n",
    "<p>In bacteria the DNA replication starts at the repliation ORI. </p>\n",
    "<p>In eukaroyates occurs along simultaneously along strand and need enzyme telomerase to close ends of \n",
    "chromosome.</p>\n",
    "<img src=\"glick5.png\">\n",
    "protein chaings are composed of peptide linked amino acids. AA are all similar structure with a \n",
    "<li>central carbon, alpha carbon</li>\n",
    "<li>amino group NH3+</li>\n",
    "<li>Hydrodgen</li>\n",
    "<li>Carboxyl, COO-</li>\n",
    "<li>R group. An R group is one of 20 groups which differentiate the 20 AA</li>\n",
    "The first AA of a protein chain has a N terminus(free amino group) for the start and a C terminus(free Carboxyl group)\n",
    "for the end. Proteins are formed from AA residues. AA residue chains when connected form a variety of shapes. DNA creates\n",
    "the RNA polynucletide chains which are different from DNA polynucleodtide chains in \n",
    "1) replace A-U pariing from A-T pariring in DNA. and 2) RNA polymerase to transcribe DNA->RNA. In bacteria only \n",
    "mRNA w/1 RNA polymerase, in eukorayotes have mRNA, tRNA, rRNA  w/3 different polynmerases. RNA polymerase looks\n",
    "binds upstream of the promoter region then starts coding the DNA to RNA using complimentatry base pair matching with\n",
    "the exception of A->U until the termination region. In an eukaryote the intons or noncoding regions are removed \n",
    "and only the exons are spliced together. Prokaroytes dont have intons. \n",
    "<img src=\"glick6.png\">\n",
    "RNA coding from DNA A-U, C-G\n",
    "<img src=\"glick6a.png\">\n",
    "The RNA polymerase synthesizes mRNA nuceotide chains from DNA starting with the promoter region. RNA Polymerase\n",
    "picks the DNA strand going from 3'->5' to start synthesis. What does the P of the start of the mRNA represent?\n",
    "<img src=\"glick6b.png\">\n",
    "A DNA gene consists of 6 parts, \n",
    "<li>1) the promoter region in the DNA where the transcription starts,</li>\n",
    "<li>2) the 5' UTR (untranslated)region</li>\n",
    "<li>3) ATG DNA start codon which maps to the mRNA start codon. Careful, mitochondrial genomes\n",
    "have different start codona. AUA/AUU in humans, GUG/UUG in prokaryote mitchondria. Prokaryotes\n",
    "have differeent start codons, 83% ecoli has AUG, 14% GUG, 3% UUG, etc.. many variations.</li>\n",
    "<li>4) DNA coding regions which maps to RNA coding region</li>\n",
    "<li>5) teh DNA stop codon</li>\n",
    "<li>6) the 3' untranslated region</li>\n",
    "Simplified bersion w/o promoter/terminator\n",
    "<img src=\"dna1.png\">\n",
    "DNA gene w/promotoer/terminator regions\n",
    "<img src=\"dna2.png\">\n",
    "\n",
    "<img src='glick6c.png'>\n",
    "The start codon always codes for methionine in eukaryotes and a modified Met (fMet) in prokaryotes. \n",
    "The most common start codon is AUG for mRNA, ATG for DNA. \n",
    "RNA polymerase reconizes the end of the DNA promoter region with the TATA box or TATA sequence and is where mRNA\n",
    "transcription starts. \n",
    "<img src=\"glick6d.png\">\n",
    "the TATA box is a noncoding element. TATA only available in eukaroytic promoters. In prokaryotes the analagous\n",
    "promoter region is a Pribnow box, TATAAT. This is an ideal consensus sequence. There is variation in this sequence. \n",
    "Also called -10 sequence because roughly 10bp upstream from start of transcripion for prokaryotes. AT bonds are easier\n",
    "to break apart than CG bonds so AT used for promoter regions where transcription starts in both prokaryotes and eukaryotes. \n",
    "\n",
    "<img src=\"tata1.png\">\n",
    "mRNA termination hairpin: \n",
    "    \n",
    "<img src=\"glick6f.png\">\n",
    "<img src=\"glick6g.png\">\n",
    "RNA Translation requires mRNA, tRNA, AA. Note the convention for adding 3' and 5' descriptors for tRNA. tRNA\n",
    "is a small sequence 70-80 NT and hairpins are allowed to form. \n",
    "<img src=\"glick6h.png\">\n",
    "3 of the NT in the middle of tRNA is an anticodon. When the tRNA folds the anticodon becomes one end and the \n",
    "other end attaches to an amino acid. \n",
    "The ribosome matches the mRNA 3 NT sequence(codon) with a corresponding tRNA opposite NT sequence. The protein is attached\n",
    "to the tRNA. \n",
    "<img src=\"glick6i.png\">\n",
    "To form a protein polymer the ribosome finds a tRNA with the anticodon by opposite base pair matching with \n",
    "the mRNA. There are 2 tRNA molecules attached to the mRNA sequence. \n",
    "<img src=\"glick6e.png\">\n",
    "The ribosome attaches first tRNA anticodon which matches the start codon on the mRNA. The second tRNA attaches and\n",
    "the first tRNA slides over one position(the tRNA upstream closer to 5' side) to bond with the second tRNA to create \n",
    "teh peptide bond. \n",
    "<img src=\"glick6j.png\">\n",
    "<img src=\"glick7.png\">\n",
    "Eukaryote gene: Showing intons being removed. After transcription the polyA tail at the end of mRNA. The right\n",
    "angle arrow depicts the start of transcription, usually after the TATA box. The arrow also indicates the DNA direction\n",
    "from 3' ->5'. Exons are 150-300 bp in length, Intons 40-10k bp long. The primary RNA has introns in them the functional\n",
    "transcript has them removed. The 3 basic phases for transcription are initiation/elongation/termination.\n",
    "<li>Initiation:RNA polymerase binds to the promoter region on DNA in 3-5 direction. </li>\n",
    "<li>Elongation:create mRNA using inverse base pairing rules. Copies all coding and noncoding regions to form primary \n",
    "transcript. The mRNA or create transcript is created from 5->3 direction</li>\n",
    "<li>Termination:</li>\n",
    "The RNA transcription process is different for eukaryotes vs. prokaryotes. There is 1 type\n",
    "of RNA polymerase for bacteria/prokaryotes. Eukaroytes have 3 different RNA polymerases. \n",
    "\n",
    "Only prokaryotes have exons and introns and have a splicing procedure to include only remove the introns in DNA. The\n",
    "introns in DNA are first transcribed to a primay transcript then removed to form a functional transcript. \n",
    "                                      \n",
    "<img src=\"glick8.png\">\n",
    "More detail on creation of mRNA and removing intons and splicing exons into mRNA showing the 2 step process of \n",
    "first removing the intons then creating the mRNA from splicing. Careful: the intron removal splicing process\n",
    "is different in different tissue sites. Sometimes exons can be skipped resulting in a different functional mRNA. \n",
    "This exon skipping procedure generates different gene products in different tissues.\n",
    "                                    \n",
    "<img src=\"glick9.png\">\n",
    "Alternative splicing of mRNA; males and females show different mRNA splicing. \n",
    "<img src=\"glick10.png\">\n",
    "M/F fruitflies produce 2 different forms of mRNA depending on sex. Careful: this causes issues w/cuttlefish\n",
    "RNA sequencing. Female FF has an intron (4) for mRNA and this stops the transcription once this intron is introduced\n",
    "while the M FF contains all 6 exons! \n",
    "<img src=\"glick11.png\">\n",
    "tRNA is 75-93 NT long. each tRNA forms a \"L\" shape. the anticodon region is roughly in the center of the tRNA and are\n",
    "UNPAIRED!!!.\n",
    "AA is attached to 3' end via a specific synthase. \"ase\" ending means this reaction requires ATP. After the AA \n",
    "    binds, the tRNA is \"charged\". \n",
    "Most of the RNA is rRNA in ribosomes. Ribsomes create the protein from tRNA/mRNA. \n",
    "transfer RNA with mRNA.\n",
    "4% is mRNA, 4% is tRNA, 90% rRNA in ribosomes.\n",
    "rna-seq measures only ribosomes? will the 4% even show up? \n",
    "<img src=\"glick11a.png\">\n",
    "See reaction rates reading for more explanation. Transcription to mRNA and translation from mRNA to proteins occur\n",
    "at same time. Protein synthesis does not wait for transcription to complete. \n",
    "<img src=\"glick12.png\">\n",
    "<p>In prokaryotes translation and transcription occur inside cell nucleus and are NOT spatially separated. The \n",
    "transcription and translation occur in parallel. </p>\n",
    "<p>In eukaryotes translation and transcription are separated. Transcriptin inside the cell nucleus, translation\n",
    "outside the cell wall. In Eukaryotes the mRNA leaves the \n",
    "cell nucleus and attaches to the endoplastic recticulium. IMPORTANT: before the mRNA leaves the cell, the 5' end is\n",
    "    capped with a modified guanine nucleotide and the 3' end is capped with a adenine nucleotide to form a polyA tail. \n",
    "    This capping allows Ribosomes to attach to the mRNA. The translation process includes mRNA, charged tRNA, protein factors\n",
    "    to elongate and terminate the polypeptide chain. \n",
    "</p>\n",
    "<img src=\"glick13.png\">\n",
    "Eukaryotes: Caps added to the 5' and 3' ends after the intons are removed and the mRNA is exported out of the cell.\n",
    "<img src=\"glick14.png\">\n",
    "Prokaryote translation: ribosome subunit attach to mRNA by basepairing between the Shine-Dalgarno sequence near the 5' end and the\n",
    "complementary sequence enar the 3' end  of the rRNA of the small ribosome subunit. This section of 5'-AUG-3' anticodon\n",
    "ofAUG.\n",
    "\n",
    "\n",
    "<img src=\"glick15.png\">\n",
    "Eukaryote translation: \n",
    "<img src=\"glick16.png\">\n",
    "Elongation phase of translation:                                         \n",
    "<img src=\"glick17.png\">\n",
    "Termination:                                     \n",
    "<img src=\"glick18.png\">\n",
    "                                        \n",
    "<img src=\"glick19.png\">\n",
    "<img src=\"glick20.png\">\n",
    "<h6>Regulation of mRNA transcripion in Bacteria</h6>\n",
    "A bacterial cell can't express all the genes simultaneously. The genes required for cell life are continusously\n",
    "expressed while there is a signaling system which turns on other genes. An operon is a cluster of genes transcribed\n",
    "into a single mRNA. An example of an operon is the lac operon in E Coli. The lac operon allows for lactose as an \n",
    "energy source vs. glucose. \n",
    "<img src=\"lac1.png\">\n",
    "The promoter region in E Coli has 2 DNA binding sites for RNA polymerase. There is only 1 RNA polymerase for \n",
    "E. Coli. The binding sites are recognized by the sigma factor protein of RNA polymerase which are used to start\n",
    "transcription. THis protein specifically binds RNA polymerase to the promoter region of genes in bacteria. Every \n",
    "molecule of RNA Polymerase contains a sigma unit; there are 7 sigma proteins for EColi. RNA polymerase starts\n",
    "initiation after the Pribnow box and sequence TTGGACA or about -10BP upstream fromteh initiation of transcription. \n",
    "A pribnow box and TTGGACA is recognized by sigma factor RpoD or sigma-70. There are repressors or regulatory proteins\n",
    "    which bind to operons or operator regions to prevent RNA Polymerase from binding to the promoter region. \n",
    "    <h6>How do you find time sequence of these repressor proteins? when environmental changes\n",
    "    cause changes cuttlefish mRNA.</h6> Shows RNA polymerase\n",
    "    binding to the promoter region of a bacteria; position is -35/-10 upstream from start of transcription. Operon\n",
    "    is listed as being after the promoter region. \n",
    "<img src=\"glick21.png\">\n",
    "<img src=\"glick22.png\">\n",
    "Repressor proten binds to operon preventing RNA Polymerase from attaching\n",
    "<img src=\"glick23.png\">\n",
    "<img src=\"glick24.png\">\n",
    "<img src=\"glick25.png\">\n",
    "<img src=\"glick26.png\">\n",
    "Eukaryote mRNA transcription regulation:\n",
    "Transcription factor proteins bind directly to DNA sequences, are usually <10bp and regulate transcription\n",
    "genes. The euraryote promoter consists of a TATA box, a CCAAT sequence, and a sequence of CG nucleotides that \n",
    "are -25, -75, -90 upstream of where the transcription starts. The first step is binding transcription factor TFIID\n",
    "or Tata binding protein TBP to a TATA sequence. The other transcription factors bind to the TFIID and DNA adjacent to the \n",
    "TATA box. RNA Poly 2 binds ot the transcription complex attached to the TATA box and starts transcription. The\n",
    "initiator location is the +1 bp. There are transcription accelerators hundreds or thousands of BP away from +1 bp.                             \n",
    "<img src=\"glick27.png\">\n",
    "Careful: do cuttlefish have neuronal cells? \n",
    "Some proteins bind to response elements and prevent transcription. For example, there is a class of about 18 vertebrate genes that are actively transcribed in nerve cells (neurons) and turned off in nonneuronal cells. Each of these neuron-active genes has a 24-bp response element that lies upstream of its transcription initiation site. This DNA sequence is called a neuron-restrictive silencer element (NRSE). In nonneuronal cells, a protein called neuron-restrictive silencer factor (NRSF) is synthesized, binds to each NRSE, and prevents transcription of each member of this set of genes.\n",
    "<img src=\"glick28.png\">\n",
    "\n",
    "                                        <img src=\"glick29.png\">\n",
    "<img src=\"glick30.png\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Discuss the basic features of DNA replication.\n",
    "Denature the double strands. For bacteria/eukaryotes start at ori of replication. For prokaryotes simultaneously \n",
    "on strand as it separates in the 3' direction. Attach complimentary Nucleotide starting w/double bond for AT and\n",
    "triple H bond for CG. Then the 2 alpha, beta, phospor groups cleave off as a pxxx and form a phosphodiester bond\n",
    "with one of the 5 carbons in the pentane sugar. \n",
    "2. Compare and contrast DNA and RNA.\n",
    "DNA={ATCG}, RNA={AUCG}, DNA double strand helix, RNA single strand. DNA replicate with DNA polymerase. RNA transcription\n",
    "with RNA polymerase and translation via ribosomes. DNA replicates only and encodes xRNA sequences. \n",
    "<h6>3. Describe the differences and similarities between prokaryotic\n",
    "and eukaryotic structural genes.</h6>\n",
    "prokaryotic: exon only, eukaryotic: exon + intons\n",
    "\n",
    "<h6>4. Describe the elongation phase of translation.</h6>\n",
    "Elongation during translation where the tRNA attaches to the mRNA in a 2 step process, match the AA to the \n",
    "5. Deduce the most likely DNA coding sequence for the following\n",
    "human protein: MAGGTWYQLFPRKMWNDSTLHPFILPMNVAG.\n",
    "\n",
    "    ATG-GCT-GGT-GGT ...\n",
    "    start codon(AUG(RNA)/TAC(DNA))-TACCGACCACCA-end codon(TAA)\n",
    "    \n",
    "alanine - ala - A (gif, interactive)\n",
    "arginine - arg - R (gif, interactive)\n",
    "asparagine - asn - N (gif, interactive)\n",
    "aspartic acid - asp - D (gif, interactive)\n",
    "cysteine - cys - C (gif, interactive)\n",
    "glutamine - gln - Q (gif, interactive)\n",
    "glutamic acid - glu - E (gif, interactive)\n",
    "glycine - gly - G (gif, interactive)\n",
    "histidine - his - H (gif, interactive)\n",
    "isoleucine - ile - I (gif, interactive)\n",
    "leucine - leu - L (gif, interactive)\n",
    "lysine - lys - K (gif, interactive)\n",
    "methionine - met - M (gif, interactive)\n",
    "phenylalanine - phe - F (gif, interactive)\n",
    "proline - pro - P (gif, interactive)\n",
    "serine - ser - S (gif, interactive)\n",
    "threonine - thr - T (gif, interactive)\n",
    "tryptophan - trp - W (gif, interactive)\n",
    "tyrosine - tyr - Y (gif, interactive)\n",
    "valine - val - V (gif, interactive)\n",
    "<img src=\"codon.png\">\n",
    "\n",
    "    \n",
    "6. Determine the amino acid sequence encoded by the following\n",
    "mRNA: GCGAUCGACGAUGUUUCUAAAAGUAUCUCAUCGAAAUGAGGGUUCGUAAUAGCGACCCGGGCGG.\n",
    "Alanine-IsoLeucine-Aspartic Acid-etc...\n",
    "    \n",
    "7. What is an operon? What is the biological significance of an\n",
    "operon?\n",
    "\n",
    "part of promoter region where set of common genes over multiple DNA genes. controls groups of genes. \n",
    "\n",
    "8. How is transcription initiation controlled in bacterial cells?\n",
    "1) primbox + sequence XXXXXXX; RNA polymerase matches this to start transcription. \n",
    "\n",
    "9. Describe the major DNA elements that are responsible for\n",
    "the transcription of eukaryotic structural genes.\n",
    "\n",
    "\n",
    "10. How are proteins transported across the cytoplasmic\n",
    "membrane of gram-positive bacterial cells?\n",
    "\n",
    "11. Describe the type II secretion system of gram-negative\n",
    "bacterial cells.\n",
    "\n",
    "12. How are secretory proteins processed in eukaryotic cells?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Translation and Transcription example</h6>\n",
    "https://www.youtube.com/watch?v=K5n0BMKZR_Q\n",
    "DNA->mRNA->Proteins. What/why do we need codon optimization to get proteins from bacteris?\n",
    "https://www.youtube.com/watch?v=XwTExa8rs5A\n",
    "\n",
    "Start w Double DNA string from mammal cell eukorate. Create the mRNA starting at the\n",
    "promoter region. Read the DNA promotor string from 3'->5' direction.  In a mammanial cell the mRNA synthesis\n",
    "occurs in the nucleus from 3->5 direction. The synthesized mRNA is from 5->3 direction. \n",
    "The RNA polymerase will add NTPs to the bottom strand. \n",
    "\n",
    "  \n",
    "<img src='mit1.png'>\n",
    "complimentary base pair with exception A->U. and the direction is from 5' to 3'. This mRNS is in th nucleus, \n",
    "transport it out into the cytoplasm where it will be translated by a ribosome into protein.\n",
    "<img src='mit2.png'>\n",
    "the ribosome will read the mRNA starting from 5' ditection and look for a start codon AUG. \n",
    "<img src='mit3.png'>\n",
    "start creating amino acids connected by peptide bond in groups of 3. \n",
    "<img src='mit4.png'>\n",
    "there are 3 stop codons and the one you pick is the one which is in frame or the one which matches modulo 3. the\n",
    "boundaries hab=ve to match\n",
    "<img src='mit5.png'>\n",
    "what happens in a mutaion? we get a different stop codon!! and a differenyt function. \n",
    "<img src='mit6.png'>\n",
    "<img src='mit7.png'>\n",
    "<img src='mit8.png'>\n",
    "<img src='mit9.png'>\n",
    "    \n",
    "mRNA string creates3 segment aa which are called codons. There is a start codon, XXX and end codon UAG. \n",
    "<img src=\"codon1.png\">\n",
    "    \n",
    "<h6>Plasmid Vectors</h6>\n",
    "They copy themselves when inside a cell. They hace sites for gene insertion. \n",
    "<img src=\"plasmid1.png\">\n",
    "Take the circular DNA cut it with a type 2 restriction enzyme to make it linear. \n",
    "<img src=\"plasmid2.png\">\n",
    "Enzume ECOR1 recognizes AATTC and cuts that leaving a 5' overhang CTTAA. ECORV is a blunt end cutter. PST1 cleaves and\n",
    "leaves a 3' overhang. These 3 restriction enzymes cut leaving either a 5' overhang, 3' overjang or blunt end. \n",
    "<img src=\"plasmid3.png\">\n",
    "cut dna and cut plasmid. Use enzyme DNA ligase to connect the the dna into the plasmid. \n",
    "<img src=\"plasmid4.png\">\n",
    "Ligase contains ATP and forms a covalent bond between the 2 cut strands. \n",
    "<img src=\"plasmid5.png\">\n",
    "\n",
    "<img src=\"plasmid6.png\">\n",
    "once the dna is inserted putting the dna into the bacterial cell is called transformation.\n",
    "the red/pink area is an antibiotic rsistant gene which is only present in the plasmids\n",
    "which took in the insert. \n",
    "<img src=\"plasmid7.png\">\n",
    "\n",
    "<img src=\"plasmid8.png\">\n",
    "<img src=\"plasmid9.png\">\n",
    "the ori is where the dna starts replication. the fork starts here for synthesis. \n",
    "You can find the ORI by finding GC skew. \n",
    "<img src=\"plasmid10.png\">\n",
    "there is a tradeoff between low copy and high copy plasmids inside bacteria. high copy means there are 40+ copies\n",
    "of the plasmid inthe bacteria. the bacteria replicate slower because most of the energy is used to replicste the plasmid lowing\n",
    "down the division of bacteria. this is called high metabolic load from the plasmids. low cpoy plasmids are used for gene \n",
    "expression work bc the bacteria can reopliacte faster. \n",
    "<img src=\"plasmid11.png\">\n",
    "<img src=\"plasmid12.png\">\n",
    "<img src=\"plasmid13.png\">\n",
    "<img src=\"plasmid14.png\">\n",
    "<img src=\"plasmid15.png\">\n",
    "<img src=\"plasmid16.png\">\n",
    "<img src=\"plasmid17.png\">\n",
    "<img src=\"plasmid18.png\">\n",
    "<img src=\"plasmid19.png\">\n",
    "<img src=\"plasmid20.png\">\n",
    "<h6>how to make cdna</h6>\n",
    "<img src='cdna1.png'>\n",
    "to study rna/proteins we start from rna and reverse transcribe dna; or create cdna from mrna.\n",
    "\n",
    "<img src='cdna2.png'>\n",
    "cDNA containas a snapshot of all the proteins at time of cdna creation\n",
    "<img src='cdna3.png'>\n",
    "mRNA contains codons, unterminasted regions used for control and a polyA tail on the 3' end. \n",
    "<img src='cdna4.png'>\n",
    "<img src='cdna5.png'>\n",
    "use magnetic beads with TTTTTT which bind to the poly A tail of mRNA. \n",
    "<img src='cdna6.png'>\n",
    "first step to synthesizwe cdna; anneal oligo primer to polyA tail. we are goint to use an enzyme\n",
    "reverse transcriptase which comes from a virus. \n",
    "virus uses it to convert its rna to dna. \n",
    "<img src='cdna7.png'>\n",
    "the reverse transcriptase finds the primer attached to the polya tail and will synthesize dna. \n",
    "<img src='cdna8.png'>\n",
    "<img src='cdna9.png'>\n",
    "<img src='cdna10.png'>\n",
    "<img src='cdna11.png'>\n",
    "<img src='cdna12.png'>\n",
    "<img src='cdna13.png'>\n",
    "<img src='cdna14.png'>\n",
    "<img src='cdna15.png'>\n",
    "<img src='cdna16.png'>\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Codon optimization</h6>\n",
    "When making protein factories and you want to increaase protein levels you can modify the codons\n",
    "<img src='opt1.png'>\n",
    "There are synonmous variations in the codons. There are 64 possible 4*4*4 combinations for AA but only 20. The last \n",
    "NTP frequently can be modified to change protein produs=ction levels. Synonmous mutations in the nucleotide sequence.\n",
    "<img src='opt2.png'>\n",
    "\n",
    "<img src='opt3.png'>\n",
    "<img src='opt4.png'>\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
