http://allseq.com/knowledge-bank/applications/transcriptome-sequencing/
transciptome RNA sequencing

example where they used fish gonads
http://gigadb.org/dataset/100362
 

complementary DNA (cDNA) is DNA synthesized from a single stranded RNA 

To obtain eukaryotic cDNA whose introns have been removed:

A eukaryotic cell transcribes the DNA (from genes) into RNA (pre-mRNA).
The same cell processes the pre-mRNA strands by removing introns, and adding a poly-A tail and 5â€™ Methyl-Guanine cap (this is known as post-transcriptional modification)
This mixture of mature mRNA strands is extracted from the cell. The poly-A tail of the post-transcriptional mRNA can be taken advantage of with oligo(dT) beads in an affinity chromatography assay.
A poly-T oligonucleotide primer is hybridized onto the poly-A tail of the mature mRNA template, or random hexamer primers can be added which contain every possible 6 base single strand of DNA and can therefore hybridize anywhere on the RNA (Reverse transcriptase requires this double-stranded segment as a primer to start its operation.)
Reverse transcriptase is added, along with deoxynucleotide triphosphates (A, T, G, C). This synthesizes one complementary strand of DNA hybridized to the original mRNA strand.
To synthesize an additional DNA strand, traditionally one would digest the RNA of the hybrid strand, using an enzyme like RNase H, or through alkali digestion method.
After digestion of the RNA, a single stranded DNA (ssDNA) is left and because single stranded nucleic acids are hydrophobic, it tends to loop around itself. It is likely that the ssDNA forms a hairpin loop at the 3' end.
From the hairpin loop, a DNA polymerase can then use it as a primer to transcribe a complementary sequence for the ss cDNA.
Now, you should be left with a double stranded cDNA with identical sequence as the mRNA of interest.

