{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Week 3: From implanted patterns to regulatory motifs</h6>\n",
    "Source:http://web.mit.edu/manoli/www/thesis/Chapter3.html\n",
    "    <p>If you took a random sequence of nucleotides and added a pattern 15N long, how would you find these 15N patterns?\n",
    "    We could use the frequent kmer solution where we look for a repeat pattern exhaustivly. If the kmers are all the\n",
    "    same we could find them. If we don't know what the kmer is we can still exhaustively find it if we assume this \n",
    "    is a repeating pattern in a long string. Use the frequent kmer algoritm from earlier\n",
    "    </p>\n",
    "    <img src=\"motif1.png\">\n",
    "    <img src=\"motif2.png\">\n",
    "    <img src=\"motif3.png\">\n",
    "    <img src=\"motif4.png\">\n",
    "    Now implant a pattern 10 times where there are 4 random mutations in the pattern. This is a (k,d) motif where the \n",
    "    implanted pattern has d random mutations and is of length k. This requires a different algorithm. Define an implanted\n",
    "    motif problem where the input is k,d where k is the length of the kmer and d is the distance mismatch for the kmer and reutrn\n",
    "    the kmers in the string. \n",
    "    <img src=\"motif5.png\">\n",
    "    There are 3 genes which control whether the plant is in day or night mode, CCA1, LCY and TOC1\n",
    "    <img src=\"motif6.png\">\n",
    "    These transcription factors bind to the genes. How does CCA1 know where to bind? Probably there are some hidden\n",
    "    messages which tell CCA1 where to bind. \n",
    "    <img src=\"motif7.png\">\n",
    "    <img src=\"motif8.png\">\n",
    "    We define a new algoritm to find these implanted motifs. find all k,d matches. Should we explore all 4^k kmers? \n",
    "    To see which match k,d kmer/mismatches? This exhaustive search will take too much time. \n",
    "    Instead of calculating each possible motif and calculating the d distance variants from it; we can assign a \n",
    "    score to each motif. \n",
    "    <img src=\"motif9.png\">\n",
    "    <img src=\"motif10.png\">\n",
    "    <img src=\"motif11.png\">\n",
    "    <img src=\"motif12.png\">\n",
    "    <img src=\"motif13.png\">\n",
    "    <img src=\"motif14.png\">\n",
    "    <img src=\"motif15.png\">\n",
    "    <img src=\"motif16.png\">\n",
    "    <img src=\"motif17.png\">\n",
    "    <img src=\"motif18.png\">\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Redo above w/stepik exercises</h6>\n",
    "<img src=\"stepikweek2a.png\">\n",
    "<img src=\"stepikweek2b.png\">\n",
    "Each nucleotide has .25% chance of occurring. A unique 9-mer has (.25)**9 chance of occuring out of the space\n",
    "of all 9-mers. In a string of 1000 nucleotides there are 1000-9+1 substrings 9 nucleotides generated by moving\n",
    "a sliding window of 9 starting from 0 and ending at the end of the DNA string. The probability is (.25)^9 * (1000-9+1) * 500 strings\n",
    "since there are 500 strings of 1000 nucleotides. \n",
    "<img src=\"stepikweek2c.png\">\n",
    "<img src=\"stepikweek2d.png\">\n",
    "<img src=\"stepikweek2e.png\">\n",
    "<img src=\"stepikweek2f.png\">\n",
    "<img src=\"stepikweek2g.png\">\n",
    "<img src=\"stepikweek2h.png\">\n",
    "<img src=\"stepikweek2i.png\">\n",
    "<img src=\"stepikweek2j.png\">\n",
    "<img src=\"stepikweek2k.png\">\n",
    "<img src=\"stepikweek2l.png\">\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def add_one(existing_kmer_list):\n",
    "    '''\n",
    "    input: current list of kmers, \n",
    "    output: return list of kmers 1 distance away from kmers in existing_kmer_list\n",
    "    '''\n",
    "    list_kmers = []\n",
    "    if len(existing_kmer_list==0):\n",
    "        list_kmers.append(\"A\");\n",
    "        list_kmers.append(\"T\");\n",
    "        list_kmers.append(\"C\");\n",
    "        list_kmers.append(\"H\");\n",
    "    else:\n",
    "        for kmer in existing_kmer_list:\n",
    "            last_nucleotide = kmer[len(kmer)-1:len(kmer)]\n",
    "            if(last_nucleotide=='A'):\n",
    "                list_kmer.append(kmer+\"T\")\n",
    "                list_kmer.append(kmer+\"C\")\n",
    "                list_kmer.append(kmer+\"G\")\n",
    "            elif(last_nucleotide=='T'):\n",
    "                list_kmer.append(kmer+\"A\")\n",
    "                list_kmer.append(kmer+\"C\")\n",
    "                list_kmer.append(kmer+\"G\")\n",
    "            elif(last_nucleotide=='C'):\n",
    "                list_kmer.append(kmer+\"A\")\n",
    "                list_kmer.append(kmer+\"T\")\n",
    "                list_kmer.append(kmer+\"G\")\n",
    "            elif(last_nucleotide=='G'):\n",
    "                list_kmer.append(kmer+\"A\")\n",
    "                list_kmer.append(kmer+\"T\")\n",
    "                list_kmer.append(kmer+\"C\")\n",
    "    return list_kmers\n",
    "\n",
    "def distance_one(kmer):\n",
    "    list_one_away = []\n",
    "    new_nucleotide = []\n",
    "    for i in range(0,len(kmer)):\n",
    "        for nucleotide in range(0,len(kmer)):\n",
    "            print(\"x:\",x,\" nucleotide\",nucleotide)\n",
    "            if (i==nucleotide):\n",
    "                #replace with 1 different so there are 3 new kmers 1 d away\n",
    "                if nucleotide[] ==\"A\":\n",
    "                    #have to iterate through a list and add 1 character each kmer in nucleotide\n",
    "                    new_nucleotide.append(\"T\")\n",
    "                    new_nucleotide.append(\"G\")\n",
    "                    new_nucleotide.append(\"C\")\n",
    "                    \n",
    "                elif nucleotide[] == \"T\"\n",
    "                    new_nucleotide.append(\"A\")\n",
    "                elif nucleotide[] == \"C\":\n",
    "                    new_nucleotide.append(\"G\")\n",
    "                elif nucleotide == \"G\":\n",
    "                    new_nucleotide.append(\"C\")\n",
    "            else:\n",
    "                #append only\n",
    "               "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def findKmersInDNA(DNA,k):\n",
    "    '''\n",
    "    input: DNA string, integer k\n",
    "    output: set of all kmers in DNA string\n",
    "    '''\n",
    "    kmer_set = set()\n",
    "    for x in range(0,len(DNA)-k+1):\n",
    "        poasible_kmer = DNA[x:x+k]\n",
    "        kmer_set.add(possible_kmer)\n",
    "    return kmer_set\n",
    "\n",
    " \n",
    "        \n",
    "def patterns_prime(kmer,d):\n",
    "    '''\n",
    "    kmer input: single kmer \n",
    "    d: integer for hamming distance from kmer\n",
    "    output: return set of all kmers distance d from kmer\n",
    "    '''\n",
    "    distance_away_kmers = set()\n",
    "    #permutation pattern, subtract after creating set of distance 1. Then add to list. \n",
    "    while(d>0):\n",
    "        #generate list for d=1 first then subtract\n",
    "        \n",
    "        \n",
    "        \n",
    "        d -=1\n",
    "    \n",
    "def motif_enumeration(DNA,k,d):\n",
    "    '''\n",
    "    k = integer of kmer length for possible kmers in DNA\n",
    "    d = hamming distance from kmer in DNA for possible match in DNA\n",
    "    DNA = input string\n",
    "    '''\n",
    "    patterns = findKmersInDNA(DNA,k)\n",
    "    for pattern in patterns_prime(kmer,d):\n",
    "        \n",
    "        \n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DNA=[\n",
    "    \"\",\n",
    "    \"\",\n",
    "    \"\",\n",
    "    \"\",\n",
    "    \"\"\n",
    "]\n",
    "k=3\n",
    "d=1\n",
    "stuff = motif_enumeration(DNA,k=3,d=1)\n",
    "print(stuff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0\n",
      "0.0\n"
     ]
    }
   ],
   "source": [
    "#calculate entropy defined as -sum(p(x)*log(p(x))) there are no 0 values b/c it throws the calculation off\n",
    "#and in the language models we add a kludge to make sure there are no 0 probabilities for words\n",
    "\n",
    "import numpy as np\n",
    "\n",
    "p1=1 #problem if p1=0\n",
    "print(np.log(p1))\n",
    "\n",
    "res = np.multiply(p1,np.log(p1))\n",
    "print(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no_zeros: [0.5 0.5]\n",
      "-0.6931471805599453\n",
      "no_zeros: [0.25 0.25 0.25 0.25]\n",
      "-1.3862943611198906\n",
      "no_zeros: [1]\n",
      "0.0\n",
      "no_zeros: [0.25 0.5  0.25]\n",
      "-1.0397207708399179\n"
     ]
    }
   ],
   "source": [
    "#http://web.mit.edu/16.unified/www/FALL/thermodynamics/notes/node57.html\n",
    "    \n",
    "#in the statistical definition of randomness and entropy the system with a uniform probability distribution \n",
    "#has the highest entropy. \n",
    "\n",
    "def entropy(a):\n",
    "    #remove 0s first then with the new array\n",
    "    b=a[a!=0]\n",
    "    print(\"no_zeros:\",b)\n",
    "    entropy = np.dot(b,np.log(b))\n",
    "    print(entropy)\n",
    "\n",
    "#you have to manually take out the zeros? \n",
    "a = np.array([.5,0.,0,.5])\n",
    "entropy(a)\n",
    "b = np.array([0.25, 0.25, 0.25, 0.25])\n",
    "entropy(b)\n",
    "c=np.array([0, 0, 0, 1])\n",
    "entropy(c)\n",
    "d = np.array([0.25, 0, 0.5, 0.25])\n",
    "entropy(d)\n",
    "\n",
    "#by inspection\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing GGGG\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'hamming' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-2f14b52088b3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     24\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mdist_kmer\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 26\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0md_kmer_collkmer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mPattern\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mMotifs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     27\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mpattern_text\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpattern\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtext\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-4-2f14b52088b3>\u001b[0m in \u001b[0;36md_kmer_collkmer\u001b[0;34m(kmer, set_kmers)\u001b[0m\n\u001b[1;32m     20\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mmotif_kmer\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mset_kmers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"processing\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmotif_kmer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m         \u001b[0md\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhamming\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkmer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmotif_kmer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m         \u001b[0mdist_kmer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0md\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mdist_kmer\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'hamming' is not defined"
     ]
    }
   ],
   "source": [
    "# there are 4 distance measures \n",
    "#1) hamming distance which we already have\n",
    "#2) d(pattern, motifs)\n",
    "#3) min dist(pattern,text)\n",
    "#4) sum dist(pattern,set of DNA)\n",
    "import numpy as np\n",
    "\n",
    "\n",
    "Text = \"TTACCATGTA\"\n",
    "Pattern = \"ACGT\"  \n",
    "Pattern_prime = \"ATAT\"\n",
    "Motifs = {\"TACG\", \"GCGT\", \"GGGG\"}\n",
    "\n",
    "def d_kmer_collkmer(kmer, set_kmers):\n",
    "    \"\"\"\n",
    "    distnace between kmer and set of kmers where k same length between \n",
    "    kmer and set of kmers. Also known as d(pattern, motifs) where pattern = kmer\n",
    "    and motifs = set{kmers}. Note: set iterators aren't consecutive like lists\n",
    "    \"\"\"\n",
    "    dist_kmer=[]\n",
    "    for motif_kmer in set_kmers:\n",
    "        print(\"processing\",motif_kmer)\n",
    "        d = hamming(kmer, motif_kmer)\n",
    "        dist_kmer.append(d)\n",
    "    return dist_kmer\n",
    "\n",
    "print(d_kmer_collkmer(Pattern,Motifs))\n",
    "\n",
    "def pattern_text(pattern, text):\n",
    "    \"\"\"\n",
    "    distance is minimum between pattern and any kmer of the text\n",
    "    \"\"\"\n",
    "    dist = []\n",
    "    for x in range(0,len(text)-len(pattern)):\n",
    "        kmer = text[x:(x+len(pattern))]\n",
    "        dist.append(hamming(pattern,kmer))\n",
    "    print(\"dist list:\",dist)\n",
    "    return np.min(dist)\n",
    "\n",
    "print (\"pattern text distance return min dist of all kmers in text\")\n",
    "print (pattern_text(Pattern,Text)) \n",
    "\n",
    "def pattern_dna(pattern,set_dna):\n",
    "    \"\"\"\n",
    "    distance is the sum of hamming diatances between pattern, dna\n",
    "    right align pattern to dna and compute hamming difference. \n",
    "    or do you find the min distance between the set of all kmers when the stirng is longer\n",
    "    \"\"\"\n",
    "    #trim dna to the same length and return sum of hamming distances\n",
    "    d =0\n",
    "    for dna in set_dna:\n",
    "        d += hamming(pattern,dna[0:len(pattern)])\n",
    "    return d\n",
    "\n",
    "Dna = {\"AGACCAGC\", \"ACGTT\", \"CACCCATGTCGG\"}\n",
    "print(\"pattern dna, return sum of all distances\")\n",
    "print (pattern_dna(\"ATGT\",Dna))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
