{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Protein Eng(2)</h6>\n",
    "Protein engineering continued Tenants of directed evolution and workflow: \n",
    "    1.Sequence diversification a.Point mutations through error-prone PCR \n",
    "    \u0002b.TODAY: Family shuffling/molecular breeding \n",
    "    \u00022.Test protein for function a.Selection \n",
    "    \u0002b.Screen \n",
    "    \u00023.Tagging/Sequence identification \n",
    "    a.In test tubes \n",
    "    \u0002b.In cells \n",
    "    \u0002c.TODAY: In droplets àiterate!\n",
    "        \u0002Protein Engineering Example \n",
    "    2:Evolution of a b-lactamase\n",
    "        \u0001using DNA shuffling à\n",
    "        Increase chances of successby building off of nature’s evolved diversityà\n",
    "        Backcrossing to learn about residues that contribute to the evolved protein function\n",
    "    5 Challenges of directed evolution\n",
    "    1. You get what you screen for (how to confirm activity?)\n",
    "    2. Dynamic range of assay\n",
    "    3. In vivo selection is fastest/highest throughput (but most difficult to set up)\n",
    "    4. In vitro screen is most reliable (but slowest, most labor intensive)\n",
    "    5. Importance of keeping DNA sequence with the protein \n",
    "    <b>Alternative methodsfor step 2 (assay) and step 3 (sequence ID):</b>\n",
    "        flow cytometry (e.g. FACS)\n",
    "        droplet microfluidics\n",
    "        microcappilary arrays(e.g. work from Cochran lab)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "3 approaches to protein engineering, taking an AA sequence and modifying it to get a particular function\n",
    "<li>Rational Design; you have idea of structure of AA/protein, how it works compared to gene and how modifications\n",
    "will create outcomes. Usually replicating published work. </li>\n",
    "<li>Directed Evolution; mimic process in nature where you have random mutations in nature which survive. Direct \n",
    "this process using specific assays. Add in selection/xxx</li>\n",
    "<li>Synthetic; design from scratch. </li>\n",
    "<h6>Focus on 2 for this lecture; Directed Evolution workflow</h6>\n",
    "<li>1) starting step is create some diversity, need a starting DNA sequence close to where you want to go. </li>\n",
    "<li>2) convert the DNA to protein, selection or screening. How to you come up with some test\n",
    "or assay? From xx lab there were 2 assays from Arnold lab; 1) in vitro use peptide to quantitively get specific\n",
    "kinetics constants(paper?), and 2) pseudo in vivo, where protein is secreted from cell. </li>\n",
    "<li>3) sequence ID, associate the protein w/the cell where it came from and you have the DNA that corresponds to the\n",
    "protein you screened/selected for. </li>\n",
    "<li>4) iterate and do diversity again</li>\n",
    "\n",
    "<h6>Screen vs Selection</h6>\n",
    "if we screen which proteins do we get? All of them. In arnolds paper\n",
    "for substilim are we screening or selection? Looking for opaque in cell. We are screening b/c we\n",
    "aren't eliminating clones.\n",
    "Screen In vitro: If not in cell, how can we get info about protein? What methods can we use here?What do they do w/peptide?\n",
    "    Lyse cell, use flourescencse. Could also use mass spec. Most laborous. Bottleneck is protein purification. Need to keep sequenece\n",
    "    and protein correlated. Most quantative information about protein. Can do at 1 protein /second for 24hrs. Can see 8x10**4. have to isolate\n",
    "    each protein. Have to look at each protein. \n",
    "Screen in Vivo: Only gathering info about fitness or cell survivial. how nature works. Plate with opaque or clear is here, the \n",
    "    protein is secreted. Can survey 10**4 caddidates; dont need to worry about protein purifiucaion.  \n",
    "Selection In vitro:Skop. \n",
    "Selection in Vivo: Casein clearing in cell \n",
    "    \n",
    "    fitness or cell surfival. what was one of the challenges in couping selection\n",
    "    in vivo. Developing subtilimsim which is going to cleave pepetide and coupel w ability for cell\n",
    "    to grow. Cell is also evolving in background what has nothing to do w/evolve. A new mechanism\n",
    "    may pop up which allows cell to survive. \n",
    "\n",
    "Subtilisin example there was 300 mutants surveyed and they got hit then they did one more cycle. 10**4 candidates. Arnold\n",
    "needed only order of 1k. Why so small? Why in other papers need so see so many. They chose screen that was close. The sequence \n",
    "space was only a couple changes in AA away. Most recent papers are much different. Bigger and bigger jumps with more recent papers.\n",
    "\n",
    "<h6>second examp[le, cleave antibiotic. Second way of creating diversith which increases chances of succes</h6>\n",
    "Cephalosporinase enzyme which cleaves antibiotic, beta-lactomase. The enzyme cleaves an amide bond. ; comes from Stemmer paper Nature 1998. \n",
    "Change substrate into O from S. Starting point was not 1 sequence but 4 related already evolved, Citrobacter, Enterobacter, Yersinaia and Klebsiella. When\n",
    "                 we start w 4 we increase diversity. The sequences are similar. All catalyze teh same reaction. Figure from paper. \n",
    "<img src=\"stemmer1.png\">\n",
    "Teh 4 are called homologs. B-lactamase homologs. These 4 sequences are likely related to each other\n",
    "evolutionary. Some parent sequence over time which has diverged. Family shuffling. or DNA shuffling. \n",
    "                 \n",
    "<img src=\"stemmer2.png\">\n",
    "PCR gene assembly. The sequences are homolgous to each other bc they are similar. If you do PCR\n",
    "                 you can anneal together sequences which came from different parents. Then can use error prone PCR\n",
    "                 to amplify the rest of the sequences and add random mutations. Called molecular breeding\n",
    "                 where you get new mutations and you get chimeras. Why is this helpful for directed evolution? \n",
    "                 there is a graph they use where they get 4 quadrants. \n",
    "<img src=\"stemmer3.png\">      \n",
    "<img src=\"stemmer4.png\">\n",
    "                 what assay did they use? Start w library of clones. Then test by cell death from\n",
    "                 antibiotic resistance. Plate these out and pick out a living colony cause it is bigger. \n",
    "                 Then sequence the plasmid which one was able to survive. So this is selection. q) can I use a gel\n",
    "                 to separate living from dead cells? \n",
    "<h6>difference between 2 examples in stemmer paper is one started w one parent, other 4 parents. </h6>\n",
    "                 \n",
    "                 \n",
    "<h6>issues</h6> \n",
    "5 major issues:\n",
    "<li>1) get what you screen for. You can get another evolution/mutation causing survival. HYow to confirm activity.\n",
    " Couple activaity of enzyme with fitness of cell. How to confirm activtiy? What might I have to do to go back in if new sequence ia \n",
    "                 enabling cell to survive screen, for example in antibiotic. Take protein and test in vitro for acctivity. Another: take DNA sequence\n",
    "                 and put it in a new parent strain, retransofrm plasmid</li>\n",
    "<li>2) Dynamic range of assay. HOw can I evolve enzymes; you want small increments. Difficult to measure. We need a coupel rounds and how to detect a little improvement. \n",
    "                 Arnold had to do kinetic measurements in vitro. </li>\n",
    "<li>3)In vivo selection, fastest. Fitness based. How to get around this. HOw to increase rate\n",
    "    we can test clones, we dont have to limit to fitness test. People doing using florescense. \n",
    "    FACS, use cells rather have human look at each one, use flow cytometry for flourescense.\n",
    "    Sometimes hard to test in context of cell.</li>\n",
    "<img src=\"facs1.png\">\n",
    "<li>4)In vitro screen, Sometimes hard to test in context of cell. How to keep DNA w/protein. Arnold recent papers she creates protein not evolving in nature. Difficult in cell. Most reliable/flexible. Problem is hardest to do\n",
    "                 labor intensive. Have to isolate and keep associated w/sequence. Limits size of library. microfluidics makes this faster. </li>\n",
    "                 <img src='fluid1.png'>\n",
    "<li>5)assay developemtn which couples DNA w/ protein expression. Can do this in a microdroplet</li>\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Crameri-Nature-DNA shuffling for directed evolution</h6>\n",
    "Powerful process and can be more effective than random point mutations generated from\n",
    "error prone PCR. \n",
    "<li>Create variations of enzyme moxalactamase using 2 approaches,use error prone PCR to create \n",
    "single point mutations in beta-lactamase gene or use DNA shuffling to produce chimeric moxalactamase. Paper showed 8x improvement in antibiotic moxalatam resistance with random poitn mutations in a single gene\n",
    "vs. 279-540 fold improvement with point mutations vs. random cutting and combining of 4 genes. Note: the key in expanding the search space is the combination of the different\n",
    "    genes which they randomly cut using some method to create DSB in the genes. They used ug/ml concentrations\n",
    "as comparison. How many ug of surviving ecoli in solution. Sufffling\n",
    "allows mixing genes from different species. Iterations of shuffling followed by screening/selection\n",
    "allows evolution for enhanced gene activity, imporved protein folding or altered substrate\n",
    "specifity. </li>\n",
    "The first figure shows 4 genes with similar BP/NT content. \n",
    "<img src=\"shuffle1.png\">\n",
    "The second figure shows multiple gene recombination. There were 2 rounds of shuffling\n",
    "on the first left path and on the second round they observed 3.5x improvement over the 8x in the \n",
    "first round. This still isnt as big as 540x. Gray segments show homology. The other benefit from \n",
    "the family shuffling approach was clone A and B were teh most resistant after the first round\n",
    "of family shuffling. Clone A was also resistant to cefoxitine, carbenicillin and cephaflouridine; 4x\n",
    "improvement over wild type. This shows new functionality. Clones A/B produced enzymes which cant \n",
    "be compared w/wild type because wild type doesnt produce enough cloned enzyome to compare with. \n",
    "\n",
    "<img src='shuffle2.png'>\n",
    "computer model of chimera clone A was created by starting w/reference protein/enzume from known\n",
    "Enterobacteria cloacae structure then adding in mutations. Clone A has 37% amino acids which\n",
    "are differnt, after energy minimization(??) the alpha chain backbone of the A chimera is almost\n",
    "identical. The block shuffling nature of family shuffling creates more mutations in many more\n",
    "positions than single random point mutations. A gingle pgene shuffled 3 times created 4 AA mutations(a lot of the\n",
    "random mutations dont create new product/enzymes ) whereas a single cycle from 4 different species\n",
    "resulted in a mutatnt enzyme w/102-196 AA difference between the 4 different species. The figure below\n",
    "ahows difference in search space for the same number of samples. Single point mutations are all clustered\n",
    "in same area, dna shuffled w/family of genes shows larger diversity and a larger search space. \n",
    "Use error prone PCR to generate random mutations and use PCR to do homology. Joining similar sequences.\n",
    "<img src='shuffle3.png'>\n",
    "Cover a lot more space in library when we do DNA/family shuffling. \n",
    "<img src='shuffle4.png'>\n",
    "<li>AmpC β-lactamases are clinically important cephalosporinases encoded on the chromosomes of many of the Enterobacteriaceae and a few other organisms, where they mediate resistance to cephalothin, cefazolin, cefoxitin, most penicillins, and β-lactamase inhibitor-β-lactam combinations. In many bacteria, AmpC enzymes are inducible and can be expressed at high levels by mutation</li>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>How to use DE to learn something new</h6>\n",
    "Once you get to new sequence take subtilisin example. 1E2A 4 mutations. ONce you get an evolved \n",
    "sequence what can you learn? Of teh 4 point mutations in 1E2A how do you know which point mutation\n",
    "is really beneficial. Some mutaions are beneficial, some are null, some are deleterious \n",
    "towards final output. Can look at structure; can also test idividyually and in combination. \n",
    "If you have 4 or 10 mutations that is a lot of combinations. Have to do site directed mutagenesis. \n",
    "Arnold lab used back crossing to make this faster instead of doing exhaustive search for all the combinations.\n",
    "\n",
    "<h6>Backcrosssing</h6>\n",
    "Uses same idea from DNA or family shuffling but start w wild and final mutated strand w 4 point mutations. \n",
    "<li>Use regular PCR to combine mutated strand w/wild type strand and use probabilites</li>\n",
    "We have 1 parent sequence, so we back cross w wild type sequence. \n",
    "<img src='back1.png'>\n",
    "<li>Test all clones in screen. </li>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
