
% Default to the notebook output style

    


% Inherit from the specified cell style.




    
\documentclass[11pt]{article}

    
    
    \usepackage[T1]{fontenc}
    % Nicer default font (+ math font) than Computer Modern for most use cases
    \usepackage{mathpazo}

    % Basic figure setup, for now with no caption control since it's done
    % automatically by Pandoc (which extracts ![](path) syntax from Markdown).
    \usepackage{graphicx}
    % We will generate all images so they have a width \maxwidth. This means
    % that they will get their normal width if they fit onto the page, but
    % are scaled down if they would overflow the margins.
    \makeatletter
    \def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth
    \else\Gin@nat@width\fi}
    \makeatother
    \let\Oldincludegraphics\includegraphics
    % Set max figure width to be 80% of text width, for now hardcoded.
    \renewcommand{\includegraphics}[1]{\Oldincludegraphics[width=.8\maxwidth]{#1}}
    % Ensure that by default, figures have no caption (until we provide a
    % proper Figure object with a Caption API and a way to capture that
    % in the conversion process - todo).
    \usepackage{caption}
    \DeclareCaptionLabelFormat{nolabel}{}
    \captionsetup{labelformat=nolabel}

    \usepackage{adjustbox} % Used to constrain images to a maximum size 
    \usepackage{xcolor} % Allow colors to be defined
    \usepackage{enumerate} % Needed for markdown enumerations to work
    \usepackage{geometry} % Used to adjust the document margins
    \usepackage{amsmath} % Equations
    \usepackage{amssymb} % Equations
    \usepackage{textcomp} % defines textquotesingle
    % Hack from http://tex.stackexchange.com/a/47451/13684:
    \AtBeginDocument{%
        \def\PYZsq{\textquotesingle}% Upright quotes in Pygmentized code
    }
    \usepackage{upquote} % Upright quotes for verbatim code
    \usepackage{eurosym} % defines \euro
    \usepackage[mathletters]{ucs} % Extended unicode (utf-8) support
    \usepackage[utf8x]{inputenc} % Allow utf-8 characters in the tex document
    \usepackage{fancyvrb} % verbatim replacement that allows latex
    \usepackage{grffile} % extends the file name processing of package graphics 
                         % to support a larger range 
    % The hyperref package gives us a pdf with properly built
    % internal navigation ('pdf bookmarks' for the table of contents,
    % internal cross-reference links, web links for URLs, etc.)
    \usepackage{hyperref}
    \usepackage{longtable} % longtable support required by pandoc >1.10
    \usepackage{booktabs}  % table support for pandoc > 1.12.2
    \usepackage[inline]{enumitem} % IRkernel/repr support (it uses the enumerate* environment)
    \usepackage[normalem]{ulem} % ulem is needed to support strikethroughs (\sout)
                                % normalem makes italics be italics, not underlines
    

    
    
    % Colors for the hyperref package
    \definecolor{urlcolor}{rgb}{0,.145,.698}
    \definecolor{linkcolor}{rgb}{.71,0.21,0.01}
    \definecolor{citecolor}{rgb}{.12,.54,.11}

    % ANSI colors
    \definecolor{ansi-black}{HTML}{3E424D}
    \definecolor{ansi-black-intense}{HTML}{282C36}
    \definecolor{ansi-red}{HTML}{E75C58}
    \definecolor{ansi-red-intense}{HTML}{B22B31}
    \definecolor{ansi-green}{HTML}{00A250}
    \definecolor{ansi-green-intense}{HTML}{007427}
    \definecolor{ansi-yellow}{HTML}{DDB62B}
    \definecolor{ansi-yellow-intense}{HTML}{B27D12}
    \definecolor{ansi-blue}{HTML}{208FFB}
    \definecolor{ansi-blue-intense}{HTML}{0065CA}
    \definecolor{ansi-magenta}{HTML}{D160C4}
    \definecolor{ansi-magenta-intense}{HTML}{A03196}
    \definecolor{ansi-cyan}{HTML}{60C6C8}
    \definecolor{ansi-cyan-intense}{HTML}{258F8F}
    \definecolor{ansi-white}{HTML}{C5C1B4}
    \definecolor{ansi-white-intense}{HTML}{A1A6B2}

    % commands and environments needed by pandoc snippets
    % extracted from the output of `pandoc -s`
    \providecommand{\tightlist}{%
      \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
    \DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
    % Add ',fontsize=\small' for more characters per line
    \newenvironment{Shaded}{}{}
    \newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{\textbf{{#1}}}}
    \newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.56,0.13,0.00}{{#1}}}
    \newcommand{\DecValTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{{#1}}}
    \newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{{#1}}}
    \newcommand{\FloatTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{{#1}}}
    \newcommand{\CharTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\StringTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\CommentTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textit{{#1}}}}
    \newcommand{\OtherTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{{#1}}}
    \newcommand{\AlertTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
    \newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.02,0.16,0.49}{{#1}}}
    \newcommand{\RegionMarkerTok}[1]{{#1}}
    \newcommand{\ErrorTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
    \newcommand{\NormalTok}[1]{{#1}}
    
    % Additional commands for more recent versions of Pandoc
    \newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.53,0.00,0.00}{{#1}}}
    \newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.73,0.40,0.53}{{#1}}}
    \newcommand{\ImportTok}[1]{{#1}}
    \newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.73,0.13,0.13}{\textit{{#1}}}}
    \newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}
    \newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}
    \newcommand{\VariableTok}[1]{\textcolor[rgb]{0.10,0.09,0.49}{{#1}}}
    \newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{\textbf{{#1}}}}
    \newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.40,0.40,0.40}{{#1}}}
    \newcommand{\BuiltInTok}[1]{{#1}}
    \newcommand{\ExtensionTok}[1]{{#1}}
    \newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.74,0.48,0.00}{{#1}}}
    \newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.49,0.56,0.16}{{#1}}}
    \newcommand{\InformationTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}
    \newcommand{\WarningTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}
    
    
    % Define a nice break command that doesn't care if a line doesn't already
    % exist.
    \def\br{\hspace*{\fill} \\* }
    % Math Jax compatability definitions
    \def\gt{>}
    \def\lt{<}
    % Document parameters
    \title{cuttlefish}
    
    
    

    % Pygments definitions
    
\makeatletter
\def\PY@reset{\let\PY@it=\relax \let\PY@bf=\relax%
    \let\PY@ul=\relax \let\PY@tc=\relax%
    \let\PY@bc=\relax \let\PY@ff=\relax}
\def\PY@tok#1{\csname PY@tok@#1\endcsname}
\def\PY@toks#1+{\ifx\relax#1\empty\else%
    \PY@tok{#1}\expandafter\PY@toks\fi}
\def\PY@do#1{\PY@bc{\PY@tc{\PY@ul{%
    \PY@it{\PY@bf{\PY@ff{#1}}}}}}}
\def\PY#1#2{\PY@reset\PY@toks#1+\relax+\PY@do{#2}}

\expandafter\def\csname PY@tok@w\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\expandafter\def\csname PY@tok@c\endcsname{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.25,0.50,0.50}{##1}}}
\expandafter\def\csname PY@tok@cp\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.74,0.48,0.00}{##1}}}
\expandafter\def\csname PY@tok@k\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@kp\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@kt\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.69,0.00,0.25}{##1}}}
\expandafter\def\csname PY@tok@o\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@ow\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.67,0.13,1.00}{##1}}}
\expandafter\def\csname PY@tok@nb\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@nf\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\expandafter\def\csname PY@tok@nc\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\expandafter\def\csname PY@tok@nn\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\expandafter\def\csname PY@tok@ne\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.82,0.25,0.23}{##1}}}
\expandafter\def\csname PY@tok@nv\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\expandafter\def\csname PY@tok@no\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.53,0.00,0.00}{##1}}}
\expandafter\def\csname PY@tok@nl\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.63,0.63,0.00}{##1}}}
\expandafter\def\csname PY@tok@ni\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.60,0.60,0.60}{##1}}}
\expandafter\def\csname PY@tok@na\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.49,0.56,0.16}{##1}}}
\expandafter\def\csname PY@tok@nt\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@nd\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.67,0.13,1.00}{##1}}}
\expandafter\def\csname PY@tok@s\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@sd\endcsname{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@si\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.73,0.40,0.53}{##1}}}
\expandafter\def\csname PY@tok@se\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.73,0.40,0.13}{##1}}}
\expandafter\def\csname PY@tok@sr\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.40,0.53}{##1}}}
\expandafter\def\csname PY@tok@ss\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\expandafter\def\csname PY@tok@sx\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@m\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@gh\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\expandafter\def\csname PY@tok@gu\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\expandafter\def\csname PY@tok@gd\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\expandafter\def\csname PY@tok@gi\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\expandafter\def\csname PY@tok@gr\endcsname{\def\PY@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\expandafter\def\csname PY@tok@ge\endcsname{\let\PY@it=\textit}
\expandafter\def\csname PY@tok@gs\endcsname{\let\PY@bf=\textbf}
\expandafter\def\csname PY@tok@gp\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\expandafter\def\csname PY@tok@go\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.53,0.53,0.53}{##1}}}
\expandafter\def\csname PY@tok@gt\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.27,0.87}{##1}}}
\expandafter\def\csname PY@tok@err\endcsname{\def\PY@bc##1{\setlength{\fboxsep}{0pt}\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{\strut ##1}}}
\expandafter\def\csname PY@tok@kc\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@kd\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@kn\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@kr\endcsname{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@bp\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\expandafter\def\csname PY@tok@fm\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\expandafter\def\csname PY@tok@vc\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\expandafter\def\csname PY@tok@vg\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\expandafter\def\csname PY@tok@vi\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\expandafter\def\csname PY@tok@vm\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\expandafter\def\csname PY@tok@sa\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@sb\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@sc\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@dl\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@s2\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@sh\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@s1\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\expandafter\def\csname PY@tok@mb\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@mf\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@mh\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@mi\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@il\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@mo\endcsname{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PY@tok@ch\endcsname{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.25,0.50,0.50}{##1}}}
\expandafter\def\csname PY@tok@cm\endcsname{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.25,0.50,0.50}{##1}}}
\expandafter\def\csname PY@tok@cpf\endcsname{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.25,0.50,0.50}{##1}}}
\expandafter\def\csname PY@tok@c1\endcsname{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.25,0.50,0.50}{##1}}}
\expandafter\def\csname PY@tok@cs\endcsname{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.25,0.50,0.50}{##1}}}

\def\PYZbs{\char`\\}
\def\PYZus{\char`\_}
\def\PYZob{\char`\{}
\def\PYZcb{\char`\}}
\def\PYZca{\char`\^}
\def\PYZam{\char`\&}
\def\PYZlt{\char`\<}
\def\PYZgt{\char`\>}
\def\PYZsh{\char`\#}
\def\PYZpc{\char`\%}
\def\PYZdl{\char`\$}
\def\PYZhy{\char`\-}
\def\PYZsq{\char`\'}
\def\PYZdq{\char`\"}
\def\PYZti{\char`\~}
% for compatibility with earlier versions
\def\PYZat{@}
\def\PYZlb{[}
\def\PYZrb{]}
\makeatother


    % Exact colors from NB
    \definecolor{incolor}{rgb}{0.0, 0.0, 0.5}
    \definecolor{outcolor}{rgb}{0.545, 0.0, 0.0}



    
    % Prevent overflowing lines due to hard-to-break entities
    \sloppy 
    % Setup hyperref package
    \hypersetup{
      breaklinks=true,  % so long urls are correctly broken across lines
      colorlinks=true,
      urlcolor=urlcolor,
      linkcolor=linkcolor,
      citecolor=citecolor,
      }
    % Slightly bigger margins than the latex defaults
    
    \geometry{verbose,tmargin=1in,bmargin=1in,lmargin=1in,rmargin=1in}
    
    

    \begin{document}
    
    
    \maketitle
    
    

    
    Cuttlefish Denovo Assembly Workflow

We were presented with approximately 60G of data files from a cuttlefish
sequence. This was not the egg. The goal is to generate a reference
sequence

A sample read looks like:

This file is different from the SRR data which you download using the
sra toolkit. A sample read from this reference genome looks like. Note
there is no 8 nucleotide sequence at the end of the @ line.: 

    Illumina Background

Source:http://nextgen.mgh.harvard.edu/IlluminaChemistry.html

The cuttlefish read files are denoted by FileName.R1 and FileName.R2.
Both of those files are matched as paired reads where
FileName=Undetermined\_S0\_L00X\_001. The first step described as
starting library matches the Biocurious Library Prep class where we used
the Nextera DNA Flex Library prep kit. There was a sample sheet in the
lab which described the adapters as we were adding them. Reading a
single strand in one direction gives us reads in the R1 file and reading
the same strand in the reverse direcion as the strand is bent down into
an upside U and reversed gives us the R2 read. The lines in each file
should be correlated to the same DNA half strand.

 

    Paired Reads

From the above image flattening out the inverted u we get a sequence of
2 reads. The inverted u is the flow cell reversing the reads. The first
read is before step 6 and the second read is after step 7. The
filename\_R1 and filename\_R2 files represent these 2 reads. It does not
matter trying to find out which one is forward and which one is reverse,
both reads are reverse complements of each other.
e.g.~TTCAG\textless{}-\textgreater{}CTGAA

    Cause of bad reads

There are many causes of bad reads. A single nucleotide error from a
reference genome can be from 1)sequencer error, 2) getting a copy from 1
chromosome and a copy from another chromosome, both of those nucleotides
in a certain position may not agree and you get some variation.

There are other types of non single nucleotide bad reads such as
adapters joining together and different types of sequencer errors. Low
quality scores are considered errors. 

    Checking the data -- Do not write programs

    Running FastQC

What is a universal adapter? https://www.biostars.org/p/4798/

Run over all the data with a bash script. Don't type in commands because
you can't explain to someone what you did if there is a problem. Make
your own log system.

 

    Tried 3 cleaning tools, AfterQC, trimmomatic, BBMap

Which one is better and why?

    AfterQC

The advantage of using AfterQC is it automatically selects the
parameters for you. Since I have no experience in this field I tried
this as a first step and tried to examine the changes in the data after
running a simple form of cleaning.

 Note the parameters for trim\_front and trim\_tail are different. One
is 15/0 the other 15/7. This step can also be done with trimmomatic but
I don't understand how the sliding window works and how it is better.

    Removing Adapters/Barcodes with BBMap

The barcodes are added under file BBMap/resources/adapters.fa. This is a
comprehensive list of the Illumina adapters.

    \begin{Verbatim}[commandchars=\\\{\}]
{\color{incolor}In [{\color{incolor}1}]:} \PY{n}{Below} \PY{o+ow}{is} \PY{n}{a} \PY{n}{comparison} \PY{n}{of} \PY{n}{BBMap} \PY{n}{vs}\PY{o}{.} \PY{n}{Trimmomatic}\PY{o}{.} \PY{n}{Trimmomatic} \PY{n}{on} \PY{n}{left}\PY{p}{,} \PY{n}{BBMap} \PY{n}{on} \PY{n}{right}\PY{o}{.} \PY{n}{Adapter} \PY{n}{content}
        \PY{n}{like} \PY{n}{original} \PY{n}{after} \PY{n}{trimmomatic}\PY{o}{.} \PY{n}{no} \PY{n}{difference}\PY{o}{.} 
        \PY{k+kn}{from} \PY{n+nn}{IPython}\PY{n+nn}{.}\PY{n+nn}{display} \PY{k}{import} \PY{n}{HTML}\PY{p}{,} \PY{n}{display}
        \PY{n}{display}\PY{p}{(}\PY{n}{HTML}\PY{p}{(}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{\PYZlt{}table\PYZgt{}\PYZlt{}tr\PYZgt{}\PYZlt{}td\PYZgt{}\PYZlt{}img src=}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{L007R2\PYZus{}orig.png}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{\PYZgt{}\PYZlt{}/td\PYZgt{}\PYZlt{}td\PYZgt{}\PYZlt{}img src=}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{L007R2\PYZus{}afterqc.png}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{\PYZgt{}\PYZlt{}/td\PYZgt{}\PYZlt{}/tr\PYZgt{}\PYZlt{}/table\PYZgt{}}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}
\end{Verbatim}


    
    \begin{verbatim}
<IPython.core.display.HTML object>
    \end{verbatim}

    
    Trimmomatic

Better on quality scores, they don't include the indexes so adapters not
trimmed worse on the adapter removal than AfterQC The problem with
trimmomatic is the index adapters are not in the file!!! 

    \begin{Verbatim}[commandchars=\\\{\}]
{\color{incolor}In [{\color{incolor}2}]:} \PY{c+c1}{\PYZsh{}original on left, trimmomatic on right. Much better!!! Trimmomatic not run on original dataset. Run }
        \PY{c+c1}{\PYZsh{}after BBMAP? verfify. forgot. bbmap\PYZhy{}\PYZgt{}trimmomatic\PYZhy{}\PYZgt{}fastqc}
        \PY{k+kn}{from} \PY{n+nn}{IPython}\PY{n+nn}{.}\PY{n+nn}{display} \PY{k}{import} \PY{n}{HTML}\PY{p}{,} \PY{n}{display}
        \PY{n}{display}\PY{p}{(}\PY{n}{HTML}\PY{p}{(}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{\PYZlt{}table\PYZgt{}\PYZlt{}tr\PYZgt{}\PYZlt{}td\PYZgt{}\PYZlt{}img src=}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{L007R1\PYZus{}orig.png}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{\PYZgt{}\PYZlt{}/td\PYZgt{}\PYZlt{}td\PYZgt{}\PYZlt{}img src=}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{ L007R1\PYZus{}trimmomatic.png}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{\PYZgt{}\PYZlt{}/td\PYZgt{}\PYZlt{}/tr\PYZgt{}\PYZlt{}/table\PYZgt{}}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}
\end{Verbatim}


    
    \begin{verbatim}
<IPython.core.display.HTML object>
    \end{verbatim}

    
    \begin{Verbatim}[commandchars=\\\{\}]
{\color{incolor}In [{\color{incolor}3}]:} \PY{c+c1}{\PYZsh{}orig on left, trimmomatic trim of adapters based on included Nexterra adapters. No indexes!!!!! }
        \PY{k+kn}{from} \PY{n+nn}{IPython}\PY{n+nn}{.}\PY{n+nn}{display} \PY{k}{import} \PY{n}{HTML}\PY{p}{,} \PY{n}{display}
        \PY{n}{display}\PY{p}{(}\PY{n}{HTML}\PY{p}{(}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{\PYZlt{}table\PYZgt{}\PYZlt{}tr\PYZgt{}\PYZlt{}td\PYZgt{}\PYZlt{}img src=}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{L007R1\PYZus{}adapter\PYZus{}orig.png}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{\PYZgt{}\PYZlt{}/td\PYZgt{}\PYZlt{}td\PYZgt{}\PYZlt{}img src=}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{L007R1\PYZus{}adapter\PYZus{}trimmomatic.png}\PY{l+s+s2}{\PYZsq{}}\PY{l+s+s2}{\PYZgt{}\PYZlt{}/td\PYZgt{}\PYZlt{}/tr\PYZgt{}\PYZlt{}/table\PYZgt{}}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}
\end{Verbatim}


    
    \begin{verbatim}
<IPython.core.display.HTML object>
    \end{verbatim}

    
    \begin{Verbatim}[commandchars=\\\{\}]
{\color{incolor}In [{\color{incolor} }]:} \PY{n}{https}\PY{p}{:}\PY{o}{/}\PY{o}{/}\PY{n}{www}\PY{o}{.}\PY{n}{biorxiv}\PY{o}{.}\PY{n}{org}\PY{o}{/}\PY{n}{content}\PY{o}{/}\PY{n}{early}\PY{o}{/}\PY{l+m+mi}{2018}\PY{o}{/}\PY{l+m+mi}{03}\PY{o}{/}\PY{l+m+mi}{01}\PY{o}{/}\PY{l+m+mi}{274100}
        \PY{n}{https}\PY{p}{:}\PY{o}{/}\PY{o}{/}\PY{n}{github}\PY{o}{.}\PY{n}{com}\PY{o}{/}\PY{n}{OpenGene}\PY{o}{/}\PY{n}{fastp}
\end{Verbatim}


    Removing overrepresented sequences from paired files

https://www.biostars.org/p/279450/ Use cutadapt

    Abyss testing the bloom filter fill rate

    \begin{Verbatim}[commandchars=\\\{\}]
{\color{incolor}In [{\color{incolor} }]:} \PY{o}{\PYZlt{}}\PY{n}{h6}\PY{o}{\PYZgt{}}\PY{n}{How} \PY{n}{Assemblers} \PY{n}{Work}\PY{o}{\PYZlt{}}\PY{o}{/}\PY{n}{h6}\PY{o}{\PYZgt{}}
        \PY{o}{\PYZlt{}}\PY{n}{img} \PY{n}{src}\PY{o}{=}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{debruin1.png}\PY{l+s+s2}{\PYZdq{}}\PY{o}{\PYZgt{}}
        \PY{o}{\PYZlt{}}\PY{n}{li}\PY{o}{\PYZgt{}}\PY{n}{https}\PY{p}{:}\PY{o}{/}\PY{o}{/}\PY{n}{www}\PY{o}{.}\PY{n}{christinaboucher}\PY{o}{.}\PY{n}{com}\PY{o}{/}\PY{n}{files}\PY{o}{/}\PY{l+m+mi}{1614}\PY{o}{/}\PY{l+m+mi}{2137}\PY{o}{/}\PY{l+m+mi}{9671}\PY{o}{/}\PY{n}{deBruijn}\PY{o}{.}\PY{n}{pdf}\PY{o}{\PYZlt{}}\PY{o}{/}\PY{n}{li}\PY{o}{\PYZgt{}}
        \PY{o}{\PYZlt{}}\PY{n}{li}\PY{o}{\PYZgt{}}\PY{n}{T}\PY{o}{.} \PY{n}{C}\PY{o}{.} \PY{n}{Conway} \PY{o+ow}{and} \PY{n}{A}\PY{o}{.} \PY{n}{J}\PY{o}{.} \PY{n}{Bromage}\PY{o}{.} \PY{n}{Succinct} \PY{n}{data} \PY{n}{structures} \PY{k}{for} \PY{n}{assembling} \PY{n}{large}
        \PY{n}{genomes}\PY{o}{.} \PY{n}{Bioinformatics}\PY{p}{,} \PY{l+m+mi}{27}\PY{p}{(}\PY{l+m+mi}{4}\PY{p}{)}\PY{p}{:}\PY{l+m+mi}{479486}\PY{p}{,} \PY{l+m+mf}{2011.}\PY{o}{\PYZlt{}}\PY{o}{/}\PY{n}{li}\PY{o}{\PYZgt{}}
        \PY{o}{\PYZlt{}}\PY{n}{li}\PY{o}{\PYZgt{}}\PY{n}{R}\PY{o}{.} \PY{n}{Chikhi} \PY{o+ow}{and} \PY{n}{G}\PY{o}{.} \PY{n}{Rizk}\PY{o}{.} \PY{n}{Space}\PY{o}{\PYZhy{}}\PY{n}{efficient} \PY{o+ow}{and} \PY{n}{exact} \PY{n}{de} \PY{n}{Bruijn} \PY{n}{graph} \PY{n}{representation}
        \PY{n}{based} \PY{n}{on} \PY{n}{a} \PY{n}{Bloom} \PY{n+nb}{filter}\PY{o}{.} \PY{n}{Algorithm}\PY{o}{.} \PY{n}{Mol}\PY{o}{.} \PY{n}{Biol}\PY{o}{.}\PY{p}{,} \PY{l+m+mi}{8}\PY{p}{(}\PY{l+m+mi}{22}\PY{p}{)}\PY{p}{,} \PY{l+m+mi}{2012}\PY{o}{\PYZlt{}}\PY{o}{/}\PY{n}{li}\PY{o}{\PYZgt{}}
        \PY{o}{\PYZlt{}}\PY{n}{li}\PY{o}{\PYZgt{}}\PY{p}{,} \PY{n}{Bowe}\PY{p}{,} \PY{n}{Onodera}\PY{p}{,} \PY{n}{Sadakane} \PY{o+ow}{and} \PY{n}{Shibuya}\PY{p}{,} \PY{n}{A}\PY{o}{.} \PY{n}{Bowe}\PY{p}{,} \PY{n}{T}\PY{o}{.} \PY{n}{Onodera}\PY{p}{,} \PY{n}{K}\PY{o}{.} \PY{n}{Sadakane}\PY{p}{,} \PY{o+ow}{and} \PY{n}{T}\PY{o}{.} \PY{n}{Shibuya}\PY{o}{.} \PY{n}{Succinct} \PY{n}{de} \PY{n}{Bruijn} \PY{n}{graphs}\PY{o}{.}
        \PY{n}{In} \PY{n}{Proc}\PY{o}{.} \PY{n}{WABI}\PY{p}{,} \PY{n}{pages} \PY{l+m+mi}{225}\PY{err}{–}\PY{l+m+mi}{235}\PY{p}{,} \PY{l+m+mf}{2012.}\PY{o}{\PYZlt{}}\PY{o}{/}\PY{n}{li}\PY{o}{\PYZgt{}}
        \PY{o}{\PYZlt{}}\PY{n}{li}\PY{o}{\PYZgt{}}\PY{p}{]} \PY{n}{R}\PY{o}{.} \PY{n}{Chikhi}\PY{p}{,} \PY{n}{A}\PY{o}{.} \PY{n}{Limasset}\PY{p}{,} \PY{n}{S}\PY{o}{.} \PY{n}{Jackman}\PY{p}{,} \PY{n}{J}\PY{o}{.} \PY{n}{Simpson}\PY{p}{,} \PY{o+ow}{and} \PY{n}{P}\PY{o}{.} \PY{n}{Medvedev}\PY{o}{.} \PY{n}{On} \PY{n}{the}
        \PY{n}{representation} \PY{n}{of} \PY{n}{de} \PY{n}{Bruijn} \PY{n}{graphs}\PY{o}{.} \PY{n}{In} \PY{n}{Proc}\PY{o}{.} \PY{n}{RECOMB}\PY{p}{,} \PY{n}{pages} \PY{l+m+mi}{35}\PY{err}{–}\PY{l+m+mi}{55}\PY{p}{,} \PY{l+m+mf}{2014.}\PY{o}{\PYZlt{}}\PY{o}{/}\PY{n}{li}\PY{o}{\PYZgt{}}
        \PY{n}{The} \PY{n}{above} \PY{n}{figure} \PY{n}{represents} \PY{n}{an} \PY{n}{Eulerian} \PY{n}{walk}\PY{p}{;} \PY{n}{walking} \PY{n}{the} \PY{n}{path} \PY{n}{such} \PY{n}{that} \PY{n}{each} \PY{n}{edge} \PY{o+ow}{is} \PY{n}{only} \PY{n}{travelled} \PY{n}{once}\PY{o}{.} \PY{n}{A} \PY{n}{vertex}
        \PY{n}{represents} \PY{n}{a} \PY{n}{single} \PY{n}{kmer} \PY{n}{of} \PY{n}{length} \PY{n}{K}\PY{o}{.} \PY{n}{Build} \PY{n}{the} \PY{n}{graph}\PY{p}{,} \PY{n}{simplify} \PY{n}{removing} \PY{n}{bubbles} \PY{o+ow}{and}
        \PY{n}{subgraphs} \PY{n}{which} \PY{n}{fall} \PY{n}{below} \PY{n}{a} \PY{n}{threshold}\PY{p}{,} \PY{o+ow}{and} \PY{n}{perform} \PY{n}{a} \PY{n}{walk} \PY{n}{to} \PY{n}{produce} \PY{n}{K}\PY{o}{+}\PY{l+m+mi}{1} \PY{n}{length} \PY{n}{kmer} \PY{n}{reads} \PY{n}{producing} 
        \PY{n}{a} \PY{n+nb}{list} \PY{n}{of} \PY{n}{contigs}\PY{o}{.} \PY{n}{Contigs} \PY{n}{are} \PY{n}{assembled} \PY{n}{using} \PY{n}{overlap}\PY{o}{.} \PY{n}{The} \PY{n}{value} \PY{n}{of} \PY{n}{K} \PY{n}{has} \PY{n}{a} \PY{n}{direct} \PY{n}{impact} \PY{n}{on} \PY{n}{the} \PY{n}{quality} \PY{n}{of} \PY{n}{the} \PY{n}{graph}
        \PY{n}{Too} \PY{n}{low} \PY{n}{a} \PY{n}{K} \PY{n}{results} \PY{o+ow}{in} \PY{n}{more} \PY{n}{spurs} \PY{o+ow}{and} \PY{n}{edges} \PY{n}{which} \PY{n}{when} \PY{n}{removed} \PY{n}{prevent} \PY{n}{assembly}\PY{o}{.} \PY{n}{Too} \PY{n}{high} \PY{n}{makes} \PY{n}{the} \PY{n}{graph} \PY{n}{sparse} \PY{o+ow}{and} 
        \PY{n}{disconnected}\PY{o}{.} \PY{n}{SPAdes} \PY{n}{uses} \PY{n}{multiple} \PY{n}{K} \PY{n}{values} \PY{n}{when} \PY{n}{running} \PY{n}{assembly}\PY{o}{.} \PY{n}{Abyss} \PY{n}{uses} \PY{n}{a} \PY{n}{distributed} \PY{n+nb}{hash} \PY{n}{table} \PY{n}{which} \PY{n}{requres} \PY{l+m+mi}{336}\PY{n}{GB}
        \PY{k}{for} \PY{n}{assembling} \PY{n}{the} \PY{n}{human} \PY{n}{genome}\PY{o}{.} \PY{n}{In} \PY{l+m+mi}{2011} \PY{n}{Conway} \PY{o+ow}{and} \PY{n}{Bromage} \PY{n}{used} \PY{n}{a} \PY{n}{sparse} \PY{n}{bitvector} \PY{n}{using} \PY{l+m+mi}{32}\PY{n}{GB}\PY{p}{,} \PY{n}{Chikhi} \PY{o+ow}{and} \PY{n}{Rizk} \PY{n}{used} \PY{n}{a} \PY{n}{bloom} 
        \PY{n+nb}{filter} \PY{n}{which} \PY{n}{took} \PY{l+m+mf}{5.7}\PY{n}{GB} \PY{o+ow}{and} \PY{n}{Bowe}\PY{p}{,} \PY{n}{Onodera}\PY{p}{,} \PY{n}{Sadakane} \PY{o+ow}{and} \PY{n}{Shibuya} \PY{n}{used} \PY{n}{the} \PY{n}{Burrows} \PY{n}{Wheeler} \PY{n}{transform} \PY{n}{to} \PY{n}{derive} \PY{n}{a} \PY{n}{data} \PY{n}{structure}
        \PY{n}{using} \PY{l+m+mf}{2.5}\PY{n}{GB}\PY{o}{.} \PY{n}{R}\PY{o}{.} \PY{n}{Chikhi}\PY{p}{,} \PY{n}{A}\PY{o}{.} \PY{n}{Limasset}\PY{p}{,} \PY{n}{S}\PY{o}{.} \PY{n}{Jackman}\PY{p}{,} \PY{n}{J}\PY{o}{.} \PY{n}{Simpson}\PY{p}{,} \PY{o+ow}{and} \PY{n}{P}\PY{o}{.} \PY{n}{Medvedev} \PY{n}{used} \PY{n}{a} \PY{n}{FM} \PY{n}{index} \PY{o+ow}{and} \PY{n}{minimizers} \PY{n}{to} \PY{n}{further} \PY{n}{reduce} \PY{n}{teh} \PY{n}{space} \PY{n}{to} \PY{l+m+mf}{1.5}\PY{n}{GB} \PY{k}{for} \PY{n}{tehe} \PY{n}{NA18507} \PY{n}{dataset}\PY{o}{.} 
        \PY{n}{https}\PY{p}{:}\PY{o}{/}\PY{o}{/}\PY{n}{github}\PY{o}{.}\PY{n}{com}\PY{o}{/}\PY{n}{jts}\PY{o}{/}\PY{n}{dbgfm}
            
\end{Verbatim}


    Overlapping graph Assemblers

Source: MIT Courseware

sga w/fm indexes better performance than SOAPdeonovo(debruijn)

Given a set of kmers how to find a substring which includes all the
kmers? Concatenate them all together. How to find the shortest common
substring? Create an overlap graph and find a path where you use 1 edge
only once. This turns out to be a solution to the genome assembly
problem. The problem is there are often many paths in the graph and it
is NP complete. There is no algorthm for finding the minimum cycle path.
Trial and error. SCS algorithm: start withlist of substriings; find
lenght of overlap between each striing. The substrings are AAA AAB ABA
ABB BAA BAB BBA BBB. The weights from AAA to AAB are 2 because the
length of the overlap, AA is 2. Make edges beween all the vertexes which
have overlap and label them.

     

    Installing, build SGA

conda create -n sga python=3.6
https://github.com/open-source-parsers/jsoncpp current jsoncpp installed
version is 1.7.4 as part of ubuntu 16.04LTS Requirement for bamtools is
jsoncpp\textgreater{}1.8.0. jsoncpp 1.8.3 gets installed as
libjsoncpp.so.2
https://github.com/pezmaster31/bamtools/wiki/Building-and-installing
https://github.com/jts/sga bam compile -- cd into src directory and run
./configure sga tools are in sga/bin not all the binaries are installed
under /usr/local dc@dc-X10DAi:\textasciitilde{}\$ cat sga.sh
\#!/bin/bash

sga index -a ropebwt --no-reverse -t 56 sgastats.fastq sga preqc -t 56
sgastats.fastq \textgreater{} sgastats.preqc sga-preqc-report.py
sgastats.preqc sga/src/examples/*.preqc 

     

    Building your own assembler

http://nbviewer.jupyter.org/gist/BenLangmead/7237207

    Velvet error

    Running Abyss

On nontrimmed or adapter removed data 

    \begin{Verbatim}[commandchars=\\\{\}]
{\color{incolor}In [{\color{incolor} }]:} \PY{o}{\PYZlt{}}\PY{o}{\PYZgt{}}
\end{Verbatim}



    % Add a bibliography block to the postdoc
    
    
    
    \end{document}
