{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Protein Engineering</h6>\n",
    "<p>Directed evolutionArnold Acc. Chem. Res. 1998, 31, 125-131Tues 5/22</p>\n",
    "<p>DNA shufflingCrameri, Nature 1998, 391, 288-291Tues5/22</p>\n",
    "OK I read those but I didnt get the invivo/kinetic constant part. \n",
    "Directed evolutionArnold Acc. Chem. Res. 1998, 31, 125-131\n",
    "Tues 5/22DNA shufflingCrameri, Nature 1998, 391, 288-291Tues5/22\n",
    "\n",
    "<h6>Lecture outline over next 2 lectures after protein folding</h6>\n",
    "Protein engineering Why make new proteins? \n",
    "-Catalysis-Binding (most modern protein therapeutics!)\n",
    "-Reaction conditions (thermostability, solvent stability etc.)\n",
    "Three  approaches  for  making  new  proteins(applications,  benefits, \n",
    "limitations  and examplesof each):\n",
    "1.Rational Engineering (site-directed mutation)\n",
    "2.Directed Evolution\n",
    "3.Design from scratch (Baker, Huang)Tenants of directed evolution and workflow: \n",
    "1.Sequence diversification \n",
    "a.Point mutations through error-prone PCR \n",
    "\u0002b.Next time: Family shuffling/molecular breeding \n",
    "\u00022.Test protein for function a.Selection \n",
    "\u0002b.Screen3.Tagging/Sequence identification \n",
    "a.In test tubes \n",
    "\u0002b.In cells \n",
    "\u0002c.In droplets à\n",
    "iterate!“Rules”of directed evolution taken from Arnold review\n",
    "\u0002Protein Engineering Example \n",
    "1:Evolution of a solvent stable protease subtilisin \n",
    "\u0001à\u0001Two different methods for assaying functionà\n",
    "Key observationsàUsing  laboratory  evolution  to  probe  fundamental  features  of  \n",
    "protein  structure  and function–back-crossing(reviewed in lecture 14)\u0001\n",
    "<h6>Alternate methods of assaying and sequence ID in Directed evolution</h6>\n",
    "<li>flow cytometry, orders of magnitude faster</li>\n",
    "<li>FACS, same as flow cytometry, requres highly specific antibodies tagged with fluorescent dye for\n",
    "each gene or mutation so you can extract them out of the stream. Separates proteins. </li>\n",
    "<li>Arrays from cochran lab, https://news.stanford.edu/2015/12/07/proteins-scale-extraction-120715/. The\n",
    "    uScale system. </li>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Making proteins; MARNDPEFC, protein chains we can go in and modify the A. What do we \n",
    "want to modify? \n",
    "<li>denovo protein design, new structure</li>\n",
    "<li>Binding to antibody</li>\n",
    "<li>Stability, put in organic solvent</li>\n",
    "<li>properties</li>\n",
    "<li>catalysis</li>\n",
    "increasing difficulty as we descend in list. 3 major approaches for doing this. 1) rational \n",
    "design, or rational mutagenisis. Choosing mutations we want to make. Couple examples, insulin\n",
    ", chimeara or protein: making an antibody from mouse and programming to person. \n",
    "<img src=\"rational.png\">\n",
    "Can see simple substrate in middle and you can do many things with it. Targeting active site\n",
    "and make modificaiton. \n",
    "Can look into here and see I need some space here to add a xxx group. \n",
    "<img src=\"rational1.png\">\n",
    "here are all products you get. How to create these modifications? You can use Restriction enzyme to cut\n",
    "and then ligase 3 pieces together where one is a different base pair. Can use PCR primer to change a\n",
    "single BP. OR can use DNA synthesis. This needs development. \n",
    "what are some benefits? We can make a change and test for that change immediately. No mutant\n",
    "libraries needed! Easy to test. Hypothesis driven, if you know something about the mechanism you\n",
    "can deduce what needs to be changed. What are some drawbacks? Have to know \n",
    "ahead of time how things work, slow. NEed structure, high failure rate(arnold paper shows high failure rate. )\n",
    "\n",
    "<img src=\"rational2.png\">\n",
    "2)Directed Evolution. What is this? When youhave a particular sequence and it evolves/survives in nature. \n",
    "Can you direct this process and make it faster? Allow for random mutagenesis. Enzyme subtilisin E. Just\n",
    "takes antibodies and breaks them. Arnold lab wanted to take this enzyme and make it thermotolerant This would be \n",
    "almost impossible using rational mutagenesis. The resulting mutations are not something yuou can \n",
    "predict which would be thermally more stable. Need initial sequence. Information rich; when you make random\n",
    "changes in protein sequence you have mroe information than in rational design. Dreawback: difficult to setup. \n",
    "Advantage/disadvantage: often successful. Advantage/disadvantage: initial sequence. Advantage:information rich, Disadvantge: difficult to setup. \n",
    "            \n",
    "<img src=\"de1.png\">\n",
    "3) Design from scratch. Build a new protein which does not exist. Advantage: dont need initail starting point. \n",
    "Advantage/disadvantage: need to be able to predict structure. Difficutl to solve computationally. Disadvantage: catalysis is hard\n",
    "        getting protein to do something is hard. Leaderi sDavid baker at UofW. \n",
    "<img src=\"baker1.png\">\n",
    "Can see there is agreement between predicted on left and actual on right w/gray. \n",
    "<img src=\"baker2.png\">\n",
    "<img src=\"baker3.png\">\n",
    "<img src=\"baker4.png\">\n",
    "Can do denovo synthesis. \n",
    "<img src=\"baker5.png\">\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h6>Directed Evolution Review</h6>\n",
    "Making new enzymes; this is the most effective method so far. Example from stemmer lab, one from arnold lab\n",
    "<li>STep1: diversify DNA. 1 parent sequence. </li>\n",
    "<li>Step2: selection/screenig, an assay to get protein function</li>\n",
    "<li>Step 3: need multiple iterations, take winner from setp2, identify sequnece \n",
    "    of aasay. Protein->DNA. </li>\n",
    "<p></p>\n",
    "SEquence space is large, how large for 20AA, how many variants 20**20. \n",
    "IF 30 AA, 20**30 cause there are 20 AA. Key is start smart. Start w/seeuence which has little of\n",
    "function we need. Take random walk in directed way. Baby steps. Last point is you get waht you assay for. \n",
    "STep for testing for protein function; can be difficult to test for this. \n",
    "<h6>Diversification</h6>\n",
    "How to get diversity? HW problem on this. Start w/error PCR on sequece. Create libaray with random mutations. \n",
    "ONe becomes new starting point. Start w DS DNA. 1 in 10**9 mutation. Why not higher mutation rate? Cause we want\n",
    "baby steps. AFter 1 round in the example 30% fall out. So we lose a lot. We lose more if we have bigger steps. \n",
    "Error PCR is one approach. \n",
    "<h6>Step 2</h6>\n",
    "Testing of proteins. 1) Put in plasmid and in cell in vivo test. 2) Or lyse cell and get protein and\n",
    "test it in vitro. \n",
    "<h6>HOw to sequence?</h6>\n",
    "Sequence identify. link protein back to piece of DNA. Cells are great way to do this. Take cell amplify\n",
    "DNA w/PCR and use this. Easy to ID sequence. Challenge have ot solve problem in context of cell. Have\n",
    "to get substrate into cell. Another way is compartmalization. In test tube have to know which protein is in which\n",
    "test tube. Consider in vitro vs in vivo; numbers the difference. \n",
    "<h6>Screening</h6>\n",
    "<li>In vivo: in cell, test each candidate. cells become red, use florsence. The number of canditates is 10**5. How? All variants in \n",
    "    cells see which ones are red. </li>\n",
    "<li>In vitro: limited. 100s. monitor rxn w color change or viscosity. </li>\n",
    "You can survey all the candidates; ones which work. \n",
    "<h6>Selection</h6>\n",
    "Only get successful candidates. based on fitness or cell survival. \n",
    "<li>In vivo: 10**10 </li>\n",
    "<li>In vitro: did not cover in class.  </li>\n",
    "\n",
    "<h6>Example: Subtilsin E</h6>\n",
    "Is an alpha-protesas an enzyme which cleaves peptide bond. In the arnold paper. Good example\n",
    "of well defined protein which we have crystal structure. Goal here is to get increased tolerance\n",
    "to solvent, DMF. Not something which you bind w/enzymes. What do you want to change in enzyme? Surface polarity you can \n",
    "change, may allow protein to fold better. Protease in context of organic solvent. If increase speed of reaction\n",
    "will this make tolerance better? Yes, increase Kcat. \n",
    "1) diversity used error prone PCR, get 3bp mutations per gene. You only want 1 AA mutation. so 1 codon or 3bp mutation. \n",
    "2) Clone bSubtils, greatest chance of growth in parent strain. This protein gets secreted from ecoli. Cell w/clone and \n",
    "a variant. Lac gene for promoter. Protein is close to cell so if in plate easy to find cell next to it. Plate otu the colonies. \n",
    "This is ok, teh flow cytometry is better. \n",
    "3) screen for activity. cleaves peptides; need some readout. 2 different assays Arnold lab used. one involved use of casein or milk protein makes opaque\n",
    "on plate. if active enzyime you get clear spot. If expressed get clear. If not expressed get cloudy. ON plate. \n",
    "SEcond assay: in Test tube, invioves antibond, if active enzyme cleaves this, you get flourescent. \n",
    "Do this in solution. Why do you want 2 assays? Forst one on is qualitative. Second one is more precise. First pass\n",
    "use first one. Second pass use more accurate. Create a library of mutants, plate them w/casein on top. 300 colonies). Wait\n",
    "for expression and you get clear and cloudy spots. The clear ones are expressed. 72% have active secretion. This tells you \n",
    "how many are active. Use second one with DMF. How many halos in second one? MEasure halos. If halo \n",
    "have enzyime for casein and for DMF. 27 clones have halos. These are bigger than the wild type. \n",
    "this is the first pass; the 27. Then do multiple rounds. \n",
    "<img src=\"de2.png\">\n",
    "<img src=\"de3.png\">\n",
    "<img src=\"de4.png\">\n",
    "<img src=\"de5.png\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
